// @bibliocommons/shared-primary-language/lib/components/PrimaryLanguage.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = PrimaryLanguage;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _reactIntl = require("react-intl");

function PrimaryLanguage(_ref) {
  var primaryLanguage = _ref.primaryLanguage,
      separator = _ref.separator,
      hiddenLanguages = _ref.hiddenLanguages;

  if (primaryLanguage && !hiddenLanguages.includes(primaryLanguage)) {
    return /*#__PURE__*/_react.default.createElement("span", {
      className: "cp-primary-language"
    }, separator, /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
      id: "LANGUAGE." + primaryLanguage
    }));
  }

  return null;
}

PrimaryLanguage.defaultProps = {
  hiddenLanguages: ['eng', 'und', 'zxx'],
  separator: ' | '
};
PrimaryLanguage.propTypes = {
  primaryLanguage: _propTypes.default.string,
  hiddenLanguages: _propTypes.default.arrayOf(_propTypes.default.string),
  separator: _propTypes.default.string
};
