// @bibliocommons/analytics-event-composers/lib/ageSuitabilityEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var AGE_SUITABILITY = _constantsAnalytics.default.CATEGORIES.AGE_SUITABILITY;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_AGE_SUITABILITY = _analyticsConstants$E.ADD_AGE_SUITABILITY,
    EDIT_AGE_SUITABILITY = _analyticsConstants$E.EDIT_AGE_SUITABILITY,
    REMOVE_AGE_SUITABILITY = _analyticsConstants$E.REMOVE_AGE_SUITABILITY;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;

function composeAgeSuitabilityEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: AGE_SUITABILITY,
    gtmAction: action
  });
}

function composeAddAgeSuitabilityEvent() {
  return (0, _compose.composeAnalytics)([composeAgeSuitabilityEvent(ADD_AGE_SUITABILITY, ADD_CONTENT)]);
}

function composeEditAgeSuitabilityEvent() {
  return (0, _compose.composeAnalytics)([composeAgeSuitabilityEvent(EDIT_AGE_SUITABILITY, EDIT_ITEM)]);
}

function composeRemoveAgeSuitabilityEvent() {
  return (0, _compose.composeAnalytics)([composeAgeSuitabilityEvent(REMOVE_AGE_SUITABILITY, REMOVE_CONTENT)]);
}

var _default = {
  composeAddAgeSuitabilityEvent: composeAddAgeSuitabilityEvent,
  composeEditAgeSuitabilityEvent: composeEditAgeSuitabilityEvent,
  composeRemoveAgeSuitabilityEvent: composeRemoveAgeSuitabilityEvent
};
exports.default = _default;
