// @bibliocommons/analytics-event-composers/lib/bibMetadataEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeExcerptClickEvent = composeExcerptClickEvent;
exports.composeFullDetailsEvent = composeFullDetailsEvent;
exports.composeMARCLinkClickEvent = composeMARCLinkClickEvent;
exports.composeMetadataLinkClickEvent = composeMetadataLinkClickEvent;
exports.composeReadMoreEvent = composeReadMoreEvent;
exports.composeShowMoreEvent = composeShowMoreEvent;
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    LABELS = _constantsAnalytics.default.LABELS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;

function composeMetadataLinkClickEvent(_ref) {
  var fieldName = _ref.fieldName,
      category = _ref.category,
      gtmContainerName = _ref.gtmContainerName;
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.BIB_METADATA_LINK, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: "metadata-" + category.toLowerCase() + "-" + fieldName.toLowerCase(),
    gtmValue: 1,
    gtmContainerName: gtmContainerName
  })]);
}

function composeReadMoreEvent(_ref2) {
  var expanded = _ref2.expanded,
      gtmContainerName = _ref2.gtmContainerName;
  var labelName = expanded ? LABELS.EXPAND : LABELS.COLLAPSE;
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.BIB_READ_MORE, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: labelName,
    gtmValue: 1,
    gtmContainerName: gtmContainerName
  })]);
}

function composeShowMoreEvent(_ref3) {
  var expanded = _ref3.expanded;
  var labelName = expanded ? LABELS.EXPAND : LABELS.COLLAPSE;
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.SHOW_MORE_CLICK, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: labelName,
    gtmValue: 1,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.BIB_ABOUT
  })]);
}

function composeFullDetailsEvent() {
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.BIB_FULL_DETAILS, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.VIEW_DETAILS,
    gtmLabel: LABELS.FULL_DETAILS,
    gtmValue: 1,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.BIB_ABOUT
  })]);
}

function composeMARCLinkClickEvent() {
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.ORIGINAL_RECORD_LINK_CLICK, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: LABELS.ORIGINAL_RECORD_VIEW,
    gtmValue: 1,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.BIB_ABOUT
  })]);
}

function composeExcerptClickEvent() {
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.EXCERPT_CLICK, {
    gtmCategory: CATEGORIES.BIB_METADATA,
    gtmAction: ACTIONS.VIEW,
    gtmLabel: LABELS.READ_EXCERPT,
    gtmValue: 1,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.BIB_TRANSACT
  })]);
}

var _default = {
  composeMetadataLinkClickEvent: composeMetadataLinkClickEvent,
  composeReadMoreEvent: composeReadMoreEvent,
  composeFullDetailsEvent: composeFullDetailsEvent,
  composeMARCLinkClickEvent: composeMARCLinkClickEvent,
  composeShowMoreEvent: composeShowMoreEvent,
  composeExcerptClickEvent: composeExcerptClickEvent
};
exports.default = _default;
