// @bibliocommons/analytics-event-composers/lib/carouselEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeAllResultsEvent = composeAllResultsEvent;
exports.composeBackwardArrowEvent = composeBackwardArrowEvent;
exports.composeCoverArtEvent = composeCoverArtEvent;
exports.composeDotEvent = composeDotEvent;
exports.composeForwardArrowEvent = composeForwardArrowEvent;
exports.composeTitleEvent = composeTitleEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _excluded = ["gtmAction", "gtmLabel", "gtmValue", "gtmContainerName", "gtmContentType"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var EVENTS = _constantsAnalytics.default.EVENTS,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    LABELS = _constantsAnalytics.default.LABELS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;

function composeCarouselEvent(_ref) {
  var gtmAction = _ref.gtmAction,
      gtmLabel = _ref.gtmLabel,
      _ref$gtmValue = _ref.gtmValue,
      gtmValue = _ref$gtmValue === void 0 ? 1 : _ref$gtmValue,
      gtmContainerName = _ref.gtmContainerName,
      gtmContentType = _ref.gtmContentType,
      payload = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  return (0, _compose.composeEvent)(EVENTS.CAROUSEL_INTERACTION, _objectSpread({
    gtmCategory: CATEGORIES.CAROUSELS,
    gtmAction: gtmAction,
    gtmLabel: gtmLabel,
    gtmValue: gtmValue,
    gtmContainerName: gtmContainerName,
    gtmContentType: gtmContentType
  }, payload));
}

function composeForwardArrowEvent(payload) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent(_objectSpread({
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: LABELS.CONTAINER_FWD_ARROW
  }, payload))]);
}

function composeBackwardArrowEvent(payload) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent(_objectSpread({
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: LABELS.CONTAINER_BWD_ARROW
  }, payload))]);
}

function composeDotEvent(payload) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent(_objectSpread({
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: LABELS.CONTAINER_DOT_INDICATORS
  }, payload))]);
}

function composeAllResultsEvent(gtmContainerName) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent({
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: LABELS.CONTAINER_VIEW_ALL,
    gtmContainerName: gtmContainerName
  })]);
}

function composeTitleEvent(payload) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent(_objectSpread({
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: LABELS.CONTAINER_TITLE
  }, payload))]);
}

function composeCoverArtEvent(gtmContainerName) {
  return (0, _compose.composeAnalytics)([composeCarouselEvent({
    gtmAction: ACTIONS.CLICK_IMAGE,
    gtmLabel: LABELS.BIB_IMAGE,
    gtmContainerName: gtmContainerName,
    gtmContentType: CUSTOM_DIMENSIONS.CONTENT_TYPES.BIB
  })]);
}

var _default = {
  composeForwardArrowEvent: composeForwardArrowEvent,
  composeBackwardArrowEvent: composeBackwardArrowEvent,
  composeDotEvent: composeDotEvent,
  composeAllResultsEvent: composeAllResultsEvent,
  composeTitleEvent: composeTitleEvent,
  composeCoverArtEvent: composeCoverArtEvent
};
exports.default = _default;
