// @bibliocommons/analytics-event-composers/lib/commentsEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.REMOVE_CONTENT = exports.REMOVE_COMMENT = exports.EDIT_ITEM = exports.EDIT_COMMENT = exports.COMMENTS = exports.ADD_CONTENT = exports.ADD_COMMENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var COMMENTS = _constantsAnalytics.default.CATEGORIES.COMMENTS;
exports.COMMENTS = COMMENTS;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_COMMENT = _analyticsConstants$E.ADD_COMMENT,
    EDIT_COMMENT = _analyticsConstants$E.EDIT_COMMENT,
    REMOVE_COMMENT = _analyticsConstants$E.REMOVE_COMMENT;
exports.REMOVE_COMMENT = REMOVE_COMMENT;
exports.EDIT_COMMENT = EDIT_COMMENT;
exports.ADD_COMMENT = ADD_COMMENT;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.EDIT_ITEM = EDIT_ITEM;
exports.ADD_CONTENT = ADD_CONTENT;

function composeCommentEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: COMMENTS,
    gtmAction: action
  });
}

function composeAddCommentEvent() {
  return (0, _compose.composeAnalytics)([composeCommentEvent(ADD_COMMENT, ADD_CONTENT)]);
}

function composeEditCommentEvent() {
  return (0, _compose.composeAnalytics)([composeCommentEvent(EDIT_COMMENT, EDIT_ITEM)]);
}

function composeRemoveCommentEvent() {
  return (0, _compose.composeAnalytics)([composeCommentEvent(REMOVE_COMMENT, REMOVE_CONTENT)]);
}

var _default = {
  composeAddCommentEvent: composeAddCommentEvent,
  composeEditCommentEvent: composeEditCommentEvent,
  composeRemoveCommentEvent: composeRemoveCommentEvent
};
exports.default = _default;
