// @bibliocommons/analytics-event-composers/lib/contentAdvisoriesEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var CONTENT_NOTICES = _constantsAnalytics.default.CATEGORIES.CONTENT_NOTICES;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_CONTENT_NOTICE = _analyticsConstants$E.ADD_CONTENT_NOTICE,
    EDIT_CONTENT_NOTICE = _analyticsConstants$E.EDIT_CONTENT_NOTICE,
    REMOVE_CONTENT_NOTICE = _analyticsConstants$E.REMOVE_CONTENT_NOTICE;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;

function composeContentAdvisoriesEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: CONTENT_NOTICES,
    gtmAction: action
  });
}

function composeAddContentAdvisoriesEvent() {
  return (0, _compose.composeAnalytics)([composeContentAdvisoriesEvent(ADD_CONTENT_NOTICE, ADD_CONTENT)]);
}

function composeEditContentAdvisoriesEvent() {
  return (0, _compose.composeAnalytics)([composeContentAdvisoriesEvent(EDIT_CONTENT_NOTICE, EDIT_ITEM)]);
}

function composeRemoveContentAdvisoriesEvent() {
  return (0, _compose.composeAnalytics)([composeContentAdvisoriesEvent(REMOVE_CONTENT_NOTICE, REMOVE_CONTENT)]);
}

var _default = {
  composeAddContentAdvisoriesEvent: composeAddContentAdvisoriesEvent,
  composeEditContentAdvisoriesEvent: composeEditContentAdvisoriesEvent,
  composeRemoveContentAdvisoriesEvent: composeRemoveContentAdvisoriesEvent
};
exports.default = _default;
