// @bibliocommons/analytics-event-composers/lib/featureInCatalogEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeFeatureInCatalogClickEvent = composeFeatureInCatalogClickEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _excluded = ["text", "url", "type"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;
var CONTENT_TYPES = CUSTOM_DIMENSIONS.CONTENT_TYPES;

function mapFicTypeToContentType(type) {
  switch (type) {
    case 'RESOURCE':
      return CONTENT_TYPES.ONLINE_RESOURCE;

    case 'EVENT':
      return CONTENT_TYPES.EVENT;

    case 'BLOG_POST':
      return CONTENT_TYPES.BLOG;

    default:
      return null;
  }
}

function composeFeatureInCatalogClickEvent(_ref) {
  var text = _ref.text,
      url = _ref.url,
      type = _ref.type,
      payload = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  var gtmContentType = mapFicTypeToContentType(type);
  return (0, _compose.composeAnalytics)([(0, _compose.composeLinkEvent)(EVENTS.FEATURE_IN_CATALOG_CLICK, url, _objectSpread({
    gtmCategory: CATEGORIES.FEATURE_IN_CATALOG,
    gtmLabel: text,
    gtmValue: 1,
    gtmContentType: gtmContentType
  }, payload))]);
}

var _default = {
  composeFeatureInCatalogClickEvent: composeFeatureInCatalogClickEvent
};
exports.default = _default;
