// @bibliocommons/analytics-event-composers/lib/holdsEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeCancelHoldEvent = composeCancelHoldEvent;
exports.composePauseHoldEvent = composePauseHoldEvent;
exports.composePlaceHoldEvent = composePlaceHoldEvent;
exports.composeResumeHoldEvent = composeResumeHoldEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _shelvesEventComposer = require("./shelvesEventComposer");

var _getLabelForProvider = _interopRequireDefault(require("./getLabelForProvider"));

var _excluded = ["event", "gtmAction", "gtmValue", "gtmLabel", "metadataIds"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;

function composeHoldEvent(_ref) {
  var event = _ref.event,
      gtmAction = _ref.gtmAction,
      gtmValue = _ref.gtmValue,
      gtmLabel = _ref.gtmLabel,
      metadataIds = _ref.metadataIds,
      payload = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  return (0, _compose.composeEvent)(event, _objectSpread({
    gtmAction: gtmAction,
    gtmValue: gtmValue,
    gtmCategory: CATEGORIES.HOLDS,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.BIB_ITEM,
    gtmLabel: gtmLabel,
    metadataIds: metadataIds
  }, payload));
}

function composeCancelHoldEvent(_ref2) {
  var holdCount = _ref2.holdCount,
      metadataIds = _ref2.metadataIds,
      shelfState = _ref2.shelfState,
      provider = _ref2.provider;
  var events = [];

  if (shelfState && shelfState.toAdd) {
    events.push((0, _shelvesEventComposer.composeShelfEvent)(EVENTS.ADD_TO_SHELF, metadataIds, shelfState.name, ACTIONS.ADD_CONTENT));
  }

  events.push(composeHoldEvent({
    event: EVENTS.CANCEL_HOLD,
    gtmAction: ACTIONS.CANCEL_HOLD,
    gtmValue: holdCount,
    gtmLabel: (0, _getLabelForProvider.default)(provider),
    metadataIds: metadataIds
  }));
  return (0, _compose.composeAnalytics)(events);
}

function composePlaceHoldEvent(_ref3) {
  var metadataIds = _ref3.metadataIds,
      provider = _ref3.provider,
      holdsPosition = _ref3.holdsPosition;
  return (0, _compose.composeAnalytics)([composeHoldEvent({
    event: EVENTS.PLACE_HOLD,
    gtmAction: ACTIONS.PLACE,
    gtmValue: 1,
    holdsPosition: holdsPosition,
    metadataIds: metadataIds,
    gtmLabel: (0, _getLabelForProvider.default)(provider)
  })]);
}

function composePauseHoldEvent(_ref4) {
  var holdCount = _ref4.holdCount,
      metadataIds = _ref4.metadataIds;
  return (0, _compose.composeAnalytics)([composeHoldEvent({
    event: EVENTS.PAUSE_HOLD,
    gtmAction: ACTIONS.PAUSE_HOLD,
    gtmValue: holdCount,
    metadataIds: metadataIds
  })]);
}

function composeResumeHoldEvent(_ref5) {
  var holdCount = _ref5.holdCount,
      metadataIds = _ref5.metadataIds;
  return (0, _compose.composeAnalytics)([composeHoldEvent({
    event: EVENTS.RESUME_HOLD,
    gtmValue: holdCount,
    metadataIds: metadataIds,
    gtmAction: ACTIONS.RESUME_HOLD
  })]);
}

var _default = {
  composePlaceHoldEvent: composePlaceHoldEvent,
  composeCancelHoldEvent: composeCancelHoldEvent,
  composePauseHoldEvent: composePauseHoldEvent,
  composeResumeHoldEvent: composeResumeHoldEvent
};
exports.default = _default;
