// @bibliocommons/analytics-event-composers/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ageSuitabilityEventComposer", {
  enumerable: true,
  get: function get() {
    return _ageSuitabilityEventComposer.default;
  }
});
Object.defineProperty(exports, "analyticsComposer", {
  enumerable: true,
  get: function get() {
    return _compose.default;
  }
});
Object.defineProperty(exports, "availabilityEventComposer", {
  enumerable: true,
  get: function get() {
    return _availabilityEventComposer.default;
  }
});
Object.defineProperty(exports, "bibEventComposer", {
  enumerable: true,
  get: function get() {
    return _bibEventComposer.default;
  }
});
Object.defineProperty(exports, "bibFormatChooser", {
  enumerable: true,
  get: function get() {
    return _bibFormatChooser.default;
  }
});
Object.defineProperty(exports, "bibHoldingsEventComposer", {
  enumerable: true,
  get: function get() {
    return _bibHoldingsEventComposer.default;
  }
});
Object.defineProperty(exports, "bibMetadataEventComposer", {
  enumerable: true,
  get: function get() {
    return _bibMetadataEventComposer.default;
  }
});
Object.defineProperty(exports, "carouselEventComposer", {
  enumerable: true,
  get: function get() {
    return _carouselEventComposer.default;
  }
});
Object.defineProperty(exports, "checkoutsEventComposer", {
  enumerable: true,
  get: function get() {
    return _checkoutsEventComposer.default;
  }
});
Object.defineProperty(exports, "commentsEventComposer", {
  enumerable: true,
  get: function get() {
    return _commentsEventComposer.default;
  }
});
Object.defineProperty(exports, "contentAdvisoriesEventComposer", {
  enumerable: true,
  get: function get() {
    return _contentAdvisoriesEventComposer.default;
  }
});
Object.defineProperty(exports, "featureInCatalogEventComposer", {
  enumerable: true,
  get: function get() {
    return _featureInCatalogEventComposer.default;
  }
});
Object.defineProperty(exports, "holdsEventComposer", {
  enumerable: true,
  get: function get() {
    return _holdsEventComposer.default;
  }
});
Object.defineProperty(exports, "iOwnThisEventComposer", {
  enumerable: true,
  get: function get() {
    return _iOwnThisEventComposer.default;
  }
});
Object.defineProperty(exports, "listCarouselEventComposer", {
  enumerable: true,
  get: function get() {
    return _listCarouselEventComposer.default;
  }
});
Object.defineProperty(exports, "novelistEventsComposer", {
  enumerable: true,
  get: function get() {
    return _novelistEventsComposer.default;
  }
});
Object.defineProperty(exports, "pageViewComposer", {
  enumerable: true,
  get: function get() {
    return _pageViewComposer.default;
  }
});
Object.defineProperty(exports, "personalizedPromotionsEventComposer", {
  enumerable: true,
  get: function get() {
    return _personalizedPromotionsEventComposer.default;
  }
});
Object.defineProperty(exports, "privateNotesEventComposer", {
  enumerable: true,
  get: function get() {
    return _privateNotesEventComposer.default;
  }
});
Object.defineProperty(exports, "quotationsEventComposer", {
  enumerable: true,
  get: function get() {
    return _quotationsEventComposer.default;
  }
});
Object.defineProperty(exports, "ratingsEventComposer", {
  enumerable: true,
  get: function get() {
    return _ratingsEventComposer.default;
  }
});
Object.defineProperty(exports, "recommendationComposer", {
  enumerable: true,
  get: function get() {
    return _recommendationComposer.default;
  }
});
Object.defineProperty(exports, "relatedTitlesEventComposer", {
  enumerable: true,
  get: function get() {
    return _relatedTitlesEventComposer.default;
  }
});
Object.defineProperty(exports, "searchEventComposer", {
  enumerable: true,
  get: function get() {
    return _searchEventComposer.default;
  }
});
Object.defineProperty(exports, "seriesCardEventComposer", {
  enumerable: true,
  get: function get() {
    return _seriesCardEventComposer.default;
  }
});
Object.defineProperty(exports, "shelvesEventComposer", {
  enumerable: true,
  get: function get() {
    return _shelvesEventComposer.default;
  }
});
Object.defineProperty(exports, "similarTitlesEventComposer", {
  enumerable: true,
  get: function get() {
    return _similarTitlesEventComposer.default;
  }
});
Object.defineProperty(exports, "summariesEventComposer", {
  enumerable: true,
  get: function get() {
    return _summariesEventComposer.default;
  }
});
Object.defineProperty(exports, "tagsEventComposer", {
  enumerable: true,
  get: function get() {
    return _tagsEventComposer.default;
  }
});
Object.defineProperty(exports, "videosEventComposer", {
  enumerable: true,
  get: function get() {
    return _videosEventComposer.default;
  }
});
Object.defineProperty(exports, "whileYouWaitEventComposer", {
  enumerable: true,
  get: function get() {
    return _whileYouWaitEventComposer.default;
  }
});

var _availabilityEventComposer = _interopRequireDefault(require("./availabilityEventComposer"));

var _bibFormatChooser = _interopRequireDefault(require("./bibFormatChooser"));

var _bibEventComposer = _interopRequireDefault(require("./bibEventComposer"));

var _commentsEventComposer = _interopRequireDefault(require("./commentsEventComposer"));

var _ratingsEventComposer = _interopRequireDefault(require("./ratingsEventComposer"));

var _tagsEventComposer = _interopRequireDefault(require("./tagsEventComposer"));

var _shelvesEventComposer = _interopRequireDefault(require("./shelvesEventComposer"));

var _holdsEventComposer = _interopRequireDefault(require("./holdsEventComposer"));

var _checkoutsEventComposer = _interopRequireDefault(require("./checkoutsEventComposer"));

var _quotationsEventComposer = _interopRequireDefault(require("./quotationsEventComposer"));

var _searchEventComposer = _interopRequireDefault(require("./searchEventComposer"));

var _featureInCatalogEventComposer = _interopRequireDefault(require("./featureInCatalogEventComposer"));

var _videosEventComposer = _interopRequireDefault(require("./videosEventComposer"));

var _iOwnThisEventComposer = _interopRequireDefault(require("./iOwnThisEventComposer"));

var _similarTitlesEventComposer = _interopRequireDefault(require("./similarTitlesEventComposer"));

var _carouselEventComposer = _interopRequireDefault(require("./carouselEventComposer"));

var _whileYouWaitEventComposer = _interopRequireDefault(require("./whileYouWaitEventComposer"));

var _privateNotesEventComposer = _interopRequireDefault(require("./privateNotesEventComposer"));

var _contentAdvisoriesEventComposer = _interopRequireDefault(require("./contentAdvisoriesEventComposer"));

var _ageSuitabilityEventComposer = _interopRequireDefault(require("./ageSuitabilityEventComposer"));

var _summariesEventComposer = _interopRequireDefault(require("./summariesEventComposer"));

var _listCarouselEventComposer = _interopRequireDefault(require("./listCarouselEventComposer"));

var _recommendationComposer = _interopRequireDefault(require("./recommendationComposer"));

var _relatedTitlesEventComposer = _interopRequireDefault(require("./relatedTitlesEventComposer"));

var _pageViewComposer = _interopRequireDefault(require("./pageViewComposer"));

var _seriesCardEventComposer = _interopRequireDefault(require("./seriesCardEventComposer"));

var _bibMetadataEventComposer = _interopRequireDefault(require("./bibMetadataEventComposer"));

var _personalizedPromotionsEventComposer = _interopRequireDefault(require("./personalizedPromotionsEventComposer"));

var _bibHoldingsEventComposer = _interopRequireDefault(require("./bibHoldingsEventComposer"));

var _novelistEventsComposer = _interopRequireDefault(require("./novelistEventsComposer"));

var _compose = _interopRequireDefault(require("./compose"));
