// @bibliocommons/analytics-event-composers/lib/iOwnThisEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TOGGLE_OWNERSHIP = exports.TOGGLE_I_OWN_THIS = exports.SHELVES = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var SHELVES = _constantsAnalytics.default.CATEGORIES.SHELVES;
exports.SHELVES = SHELVES;
var TOGGLE_I_OWN_THIS = _constantsAnalytics.default.EVENTS.TOGGLE_I_OWN_THIS;
exports.TOGGLE_I_OWN_THIS = TOGGLE_I_OWN_THIS;
var TOGGLE_OWNERSHIP = _constantsAnalytics.default.ACTIONS.TOGGLE_OWNERSHIP;
exports.TOGGLE_OWNERSHIP = TOGGLE_OWNERSHIP;

function composeIOwnThisEvent(iOwnThisStatus) {
  return (0, _compose.composeEvent)(TOGGLE_I_OWN_THIS, {
    gtmValue: 1,
    gtmLabel: iOwnThisStatus,
    gtmCategory: SHELVES,
    gtmAction: TOGGLE_OWNERSHIP
  });
}

function composeToggleIOwnThisEvent(iOwnThisStatus) {
  // conversion made because gtmLabel is expecting a string
  var status = iOwnThisStatus ? 'true' : 'false';
  return (0, _compose.composeAnalytics)([composeIOwnThisEvent(status)]);
}

var _default = {
  composeToggleIOwnThisEvent: composeToggleIOwnThisEvent
};
exports.default = _default;
