// @bibliocommons/analytics-event-composers/lib/listCarouselEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var EVENTS = _constantsAnalytics.default.EVENTS,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;

function composeListCarouselEvent(_ref) {
  var gtmAction = _ref.gtmAction,
      gtmLabel = _ref.gtmLabel,
      _ref$gtmValue = _ref.gtmValue,
      gtmValue = _ref$gtmValue === void 0 ? 1 : _ref$gtmValue,
      gtmContainerName = _ref.gtmContainerName,
      gtmContentCreator = _ref.gtmContentCreator,
      gtmCreatorType = _ref.gtmCreatorType,
      gtmCardTitle = _ref.gtmCardTitle;
  return (0, _compose.composeEvent)(EVENTS.LIST_CARD, {
    gtmCategory: CATEGORIES.CARDS_LIST,
    gtmAction: gtmAction,
    gtmLabel: gtmLabel,
    gtmValue: gtmValue,
    gtmContainerName: gtmContainerName,
    gtmContentType: CUSTOM_DIMENSIONS.CONTENT_TYPES.LIST,
    gtmContentCreator: gtmContentCreator,
    gtmCreatorType: gtmCreatorType,
    gtmCardTitle: gtmCardTitle
  });
}

function composeCoverArtClickEvent(_ref2) {
  var name = _ref2.name,
      containerName = _ref2.containerName,
      contentCreator = _ref2.contentCreator,
      creatorType = _ref2.creatorType,
      gtmCardTitle = _ref2.gtmCardTitle;
  return (0, _compose.composeAnalytics)([composeListCarouselEvent({
    gtmAction: ACTIONS.CLICK_IMAGE,
    gtmLabel: name,
    gtmContainerName: containerName,
    gtmContentCreator: contentCreator,
    gtmCreatorType: creatorType,
    gtmCardTitle: gtmCardTitle
  })]);
}

function composeTitleClickEvent(_ref3) {
  var name = _ref3.name,
      containerName = _ref3.containerName,
      contentCreator = _ref3.contentCreator,
      creatorType = _ref3.creatorType,
      gtmCardTitle = _ref3.gtmCardTitle;
  return (0, _compose.composeAnalytics)([composeListCarouselEvent({
    gtmAction: ACTIONS.CLICK_TITLE,
    gtmLabel: name,
    gtmContainerName: containerName,
    gtmContentCreator: contentCreator,
    gtmCreatorType: creatorType,
    gtmCardTitle: gtmCardTitle
  })]);
}

function composeCreatorNameClickEvent(_ref4) {
  var name = _ref4.name,
      containerName = _ref4.containerName;
  return (0, _compose.composeAnalytics)([composeListCarouselEvent({
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: name,
    gtmContainerName: containerName
  })]);
}

var _default = {
  composeCoverArtClickEvent: composeCoverArtClickEvent,
  composeTitleClickEvent: composeTitleClickEvent,
  composeCreatorNameClickEvent: composeCreatorNameClickEvent
};
exports.default = _default;
