// @bibliocommons/analytics-event-composers/lib/novelistEventsComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeViewAllEvent = composeViewAllEvent;
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    LABELS = _constantsAnalytics.default.LABELS;

function composeViewAllEvent(_ref) {
  var gtmContainerName = _ref.gtmContainerName;
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.NOVELIST_VIEW_ALL_CLICK, {
    gtmCategory: CATEGORIES.BIB_EXPLORE_FURTHER,
    gtmAction: ACTIONS.VIEW,
    gtmLabel: LABELS.NOVELIST,
    gtmValue: 1,
    gtmContainerName: gtmContainerName
  })]);
}

var _default = {
  composeViewAllEvent: composeViewAllEvent
};
exports.default = _default;
