// @bibliocommons/analytics-event-composers/lib/personalizedPromotionsEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeImageClickEvent = composeImageClickEvent;
exports.composeImpressionEvent = composeImpressionEvent;
exports.composeTitleClickEvent = composeTitleClickEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _cuid = _interopRequireDefault(require("cuid"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _excluded = ["postType", "cardId", "title", "contentType", "contentCreator", "recommendationType", "requestId"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    EVENTS = _constantsAnalytics.default.EVENTS,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;
var CONTAINER_NAMES = CUSTOM_DIMENSIONS.CONTAINER_NAMES,
    CONTENT_TYPES = CUSTOM_DIMENSIONS.CONTENT_TYPES,
    CREATOR_TYPES = CUSTOM_DIMENSIONS.CREATOR_TYPES;

function composePersonalizedPromotionsEvent(eventName, _ref) {
  var postType = _ref.postType,
      cardId = _ref.cardId,
      title = _ref.title,
      contentType = _ref.contentType,
      contentCreator = _ref.contentCreator,
      recommendationType = _ref.recommendationType,
      requestId = _ref.requestId,
      rest = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  var cardTitle = title + " | " + cardId;
  var payload = {
    gtmLabel: cardTitle,
    gtmContainerName: recommendationType === 'spotlight' ? CONTAINER_NAMES.PERSONALIZED_PROMOTION_SPOTLIGHT : CONTAINER_NAMES.PERSONALIZED_PROMOTION,
    gtmContentCreator: contentCreator,
    gtmCardTitle: cardTitle,
    gtmValue: 1,
    cardId: cardId,
    postType: postType,
    recommendationType: recommendationType,
    requestId: requestId
  };

  switch (postType) {
    case 'post':
      {
        payload.gtmCategory = CATEGORIES.CARDS_BLOG;
        payload.gtmContentType = CONTENT_TYPES.BLOG;
        payload.gtmCreatorType = CREATOR_TYPES.LIBRARY_STAFF;
        break;
      }

    case 'bw_custom_card':
      {
        payload.gtmCategory = CATEGORIES.CARDS_CUSTOM;
        break;
      }

    case 'bw_event':
      {
        payload.gtmCategory = CATEGORIES.CARDS_EVENT;
        payload.gtmContentType = CONTENT_TYPES.EVENT;
        break;
      }

    case 'bw_list':
      {
        payload.gtmCategory = CATEGORIES.CARDS_LIST;
        payload.gtmContentType = CONTENT_TYPES.LIST;
        payload.gtmCreatorType = contentType === 'staff-list' ? CREATOR_TYPES.LIBRARY_STAFF : CREATOR_TYPES.LIBRARY_PATRON;
        break;
      }

    case 'bccms_news':
      {
        payload.gtmCategory = CATEGORIES.CARDS_NEWS;
        payload.gtmContentType = CONTENT_TYPES.NEWS;
        break;
      }

    case 'bw_or_card':
      {
        payload.gtmCategory = CATEGORIES.CARDS_ONLINE_RESOURCE;
        payload.gtmContentType = CONTENT_TYPES.ONLINE_RESOURCE;
        break;
      }

    default:
      {
        throw new Error("Unexpected postType " + postType);
      }
  }

  return (0, _compose.composeEvent)(eventName, _objectSpread(_objectSpread({}, payload), rest));
}

function composeImpressionEvent(_ref2) {
  var postType = _ref2.postType,
      cardId = _ref2.cardId,
      title = _ref2.title,
      contentType = _ref2.contentType,
      contentCreator = _ref2.contentCreator,
      recommendationType = _ref2.recommendationType,
      requestId = _ref2.requestId,
      position = _ref2.position;
  var impressionId = (0, _cuid.default)();
  return (0, _compose.composeAnalytics)([composePersonalizedPromotionsEvent(EVENTS.CARD_IMPRESSION, {
    gtmAction: ACTIONS.IMPRESSION,
    postType: postType,
    cardId: cardId,
    title: title,
    contentType: contentType,
    contentCreator: contentCreator,
    impressionId: impressionId,
    recommendationType: recommendationType,
    requestId: requestId,
    position: position
  })]);
}

function composeTitleClickEvent(_ref3) {
  var postType = _ref3.postType,
      cardId = _ref3.cardId,
      title = _ref3.title,
      contentType = _ref3.contentType,
      contentCreator = _ref3.contentCreator,
      recommendationType = _ref3.recommendationType,
      requestId = _ref3.requestId,
      position = _ref3.position;
  return (0, _compose.composeAnalytics)([composePersonalizedPromotionsEvent(EVENTS.CARD_CLICK, {
    gtmAction: ACTIONS.CLICK_TITLE,
    postType: postType,
    cardId: cardId,
    title: title,
    contentType: contentType,
    contentCreator: contentCreator,
    recommendationType: recommendationType,
    requestId: requestId,
    position: position
  })]);
}

function composeImageClickEvent(_ref4) {
  var postType = _ref4.postType,
      cardId = _ref4.cardId,
      title = _ref4.title,
      contentType = _ref4.contentType,
      contentCreator = _ref4.contentCreator,
      recommendationType = _ref4.recommendationType,
      requestId = _ref4.requestId,
      position = _ref4.position;
  return (0, _compose.composeAnalytics)([composePersonalizedPromotionsEvent(EVENTS.CARD_CLICK, {
    gtmAction: ACTIONS.CLICK_IMAGE,
    postType: postType,
    cardId: cardId,
    title: title,
    contentType: contentType,
    contentCreator: contentCreator,
    recommendationType: recommendationType,
    requestId: requestId,
    position: position
  })]);
}

var _default = {
  composeImpressionEvent: composeImpressionEvent,
  composeTitleClickEvent: composeTitleClickEvent,
  composeImageClickEvent: composeImageClickEvent
};
exports.default = _default;
