// @bibliocommons/analytics-event-composers/lib/privateNotesEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.REMOVE_PRIVATE_NOTE = exports.REMOVE_CONTENT = exports.PRIVATE_NOTES = exports.EDIT_PRIVATE_NOTE = exports.EDIT_ITEM = exports.ADD_PRIVATE_NOTE = exports.ADD_CONTENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var PRIVATE_NOTES = _constantsAnalytics.default.CATEGORIES.PRIVATE_NOTES;
exports.PRIVATE_NOTES = PRIVATE_NOTES;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_PRIVATE_NOTE = _analyticsConstants$E.ADD_PRIVATE_NOTE,
    EDIT_PRIVATE_NOTE = _analyticsConstants$E.EDIT_PRIVATE_NOTE,
    REMOVE_PRIVATE_NOTE = _analyticsConstants$E.REMOVE_PRIVATE_NOTE;
exports.REMOVE_PRIVATE_NOTE = REMOVE_PRIVATE_NOTE;
exports.EDIT_PRIVATE_NOTE = EDIT_PRIVATE_NOTE;
exports.ADD_PRIVATE_NOTE = ADD_PRIVATE_NOTE;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.EDIT_ITEM = EDIT_ITEM;
exports.ADD_CONTENT = ADD_CONTENT;

function composePrivateNoteEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: PRIVATE_NOTES,
    gtmAction: action
  });
}

function composeAddPrivateNoteEvent() {
  return (0, _compose.composeAnalytics)([composePrivateNoteEvent(ADD_PRIVATE_NOTE, ADD_CONTENT)]);
}

function composeEditPrivateNoteEvent() {
  return (0, _compose.composeAnalytics)([composePrivateNoteEvent(EDIT_PRIVATE_NOTE, EDIT_ITEM)]);
}

function composeRemovePrivateNoteEvent() {
  return (0, _compose.composeAnalytics)([composePrivateNoteEvent(REMOVE_PRIVATE_NOTE, REMOVE_CONTENT)]);
}

var _default = {
  composeAddPrivateNoteEvent: composeAddPrivateNoteEvent,
  composeEditPrivateNoteEvent: composeEditPrivateNoteEvent,
  composeRemovePrivateNoteEvent: composeRemovePrivateNoteEvent
};
exports.default = _default;
