// @bibliocommons/analytics-event-composers/lib/quotationsEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.REMOVE_QUOTATION = exports.REMOVE_CONTENT = exports.QUOTATIONS = exports.EDIT_QUOTATION = exports.EDIT_ITEM = exports.ADD_QUOTATION = exports.ADD_CONTENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var QUOTATIONS = _constantsAnalytics.default.CATEGORIES.QUOTATIONS;
exports.QUOTATIONS = QUOTATIONS;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_QUOTATION = _analyticsConstants$E.ADD_QUOTATION,
    EDIT_QUOTATION = _analyticsConstants$E.EDIT_QUOTATION,
    REMOVE_QUOTATION = _analyticsConstants$E.REMOVE_QUOTATION;
exports.REMOVE_QUOTATION = REMOVE_QUOTATION;
exports.EDIT_QUOTATION = EDIT_QUOTATION;
exports.ADD_QUOTATION = ADD_QUOTATION;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.EDIT_ITEM = EDIT_ITEM;
exports.ADD_CONTENT = ADD_CONTENT;

function composeQuotationEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: QUOTATIONS,
    gtmAction: action
  });
}

function composeAddQuotationEvent() {
  return (0, _compose.composeAnalytics)([composeQuotationEvent(ADD_QUOTATION, ADD_CONTENT)]);
}

function composeEditQuotationEvent() {
  return (0, _compose.composeAnalytics)([composeQuotationEvent(EDIT_QUOTATION, EDIT_ITEM)]);
}

function composeRemoveQuotationEvent() {
  return (0, _compose.composeAnalytics)([composeQuotationEvent(REMOVE_QUOTATION, REMOVE_CONTENT)]);
}

var _default = {
  composeAddQuotationEvent: composeAddQuotationEvent,
  composeEditQuotationEvent: composeEditQuotationEvent,
  composeRemoveQuotationEvent: composeRemoveQuotationEvent
};
exports.default = _default;
