// @bibliocommons/analytics-event-composers/lib/ratingsEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.REMOVE_RATING = exports.REMOVE_CONTENT = exports.RATINGS = exports.EDIT_RATING = exports.EDIT_ITEM = exports.ADD_RATING = exports.ADD_CONTENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var RATINGS = _constantsAnalytics.default.CATEGORIES.RATINGS;
exports.RATINGS = RATINGS;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_RATING = _analyticsConstants$E.ADD_RATING,
    EDIT_RATING = _analyticsConstants$E.EDIT_RATING,
    REMOVE_RATING = _analyticsConstants$E.REMOVE_RATING;
exports.REMOVE_RATING = REMOVE_RATING;
exports.EDIT_RATING = EDIT_RATING;
exports.ADD_RATING = ADD_RATING;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
/**
 * @param {string} ratingValue rating is based out of 10/100; ie 10, 30, 70 are valid values
 */

exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.EDIT_ITEM = EDIT_ITEM;
exports.ADD_CONTENT = ADD_CONTENT;

function composeRatingEvent(eventName, ratingValue, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: ratingValue,
    gtmCategory: RATINGS,
    gtmAction: action
  });
}

function composeAddRatingEvent(ratingValue) {
  return (0, _compose.composeAnalytics)([composeRatingEvent(ADD_RATING, ratingValue, ADD_CONTENT)]);
}

function composeEditRatingEvent(ratingValue) {
  return (0, _compose.composeAnalytics)([composeRatingEvent(EDIT_RATING, ratingValue, EDIT_ITEM)]);
}

function composeRemoveRatingEvent() {
  return (0, _compose.composeAnalytics)([composeRatingEvent(REMOVE_RATING, undefined, REMOVE_CONTENT)]);
}

var _default = {
  composeAddRatingEvent: composeAddRatingEvent,
  composeEditRatingEvent: composeEditRatingEvent,
  composeRemoveRatingEvent: composeRemoveRatingEvent
};
exports.default = _default;
