// @bibliocommons/analytics-event-composers/lib/recommendationComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeImageClickEvent = composeImageClickEvent;
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;
var CONTENT_TYPES = CUSTOM_DIMENSIONS.CONTENT_TYPES,
    CONTAINER_NAMES = CUSTOM_DIMENSIONS.CONTAINER_NAMES;

function composeImageClickEvent(_ref) {
  var providerURI = _ref.providerURI,
      metadataId = _ref.metadataId;
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(EVENTS.RECOMMENDATION_CLICK, {
    gtmCategory: CATEGORIES.BIB_EXPLORE,
    gtmAction: ACTIONS.CLICK_IMAGE,
    gtmValue: 1,
    gtmLabel: undefined,
    gtmContentType: CONTENT_TYPES.BIB,
    gtmContainerName: CONTAINER_NAMES.BIB_RECOMMENDATIONS,
    gtmContentCreator: providerURI,
    metadataIds: [metadataId]
  })]);
}

var _default = {
  composeImageClickEvent: composeImageClickEvent
};
exports.default = _default;
