// @bibliocommons/analytics-event-composers/lib/searchEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeBroadenSearchEvent = composeBroadenSearchEvent;
exports.composeChatWithLibrarianClickEvent = composeChatWithLibrarianClickEvent;
exports.composeDidNotFindMoreResourcesClickEvent = composeDidNotFindMoreResourcesClickEvent;
exports.composeDidNotFindPanelToggleEvent = composeDidNotFindPanelToggleEvent;
exports.composeDidNotFindResourceClickEvent = composeDidNotFindResourceClickEvent;
exports.composeDidNotFindServiceClickEvent = composeDidNotFindServiceClickEvent;
exports.composeDidYouMeanClickEvent = composeDidYouMeanClickEvent;
exports.composeEmptySearchResultsEvent = composeEmptySearchResultsEvent;
exports.composeExternalSearchClickEvent = composeExternalSearchClickEvent;
exports.composeExternalSearchPanelClickEvent = composeExternalSearchPanelClickEvent;
exports.composeInitialPageviewSearchEvent = composeInitialPageviewSearchEvent;
exports.composeModifySearchClickEvent = composeModifySearchClickEvent;
exports.composeOtherResourcesClickEvent = composeOtherResourcesClickEvent;
exports.composeSaveSearchEvent = composeSaveSearchEvent;
exports.composeSearchProbeClickEvent = composeSearchProbeClickEvent;
exports.composeSearchResultsEvent = composeSearchResultsEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _excluded = ["event", "gtmCategory", "gtmAction", "gtmValue", "gtmLabel"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS,
    LABELS = _constantsAnalytics.default.LABELS; // eslint-disable-next-line max-len

function composeSearchEvent(_ref) {
  var event = _ref.event,
      _ref$gtmCategory = _ref.gtmCategory,
      gtmCategory = _ref$gtmCategory === void 0 ? CATEGORIES.SEARCH : _ref$gtmCategory,
      gtmAction = _ref.gtmAction,
      _ref$gtmValue = _ref.gtmValue,
      gtmValue = _ref$gtmValue === void 0 ? 1 : _ref$gtmValue,
      gtmLabel = _ref.gtmLabel,
      payload = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(event, _objectSpread({
    gtmCategory: gtmCategory,
    gtmAction: gtmAction,
    gtmValue: gtmValue,
    gtmLabel: gtmLabel
  }, payload))]);
}

function composeBroadenSearchEvent(_ref2) {
  var query = _ref2.query,
      searchType = _ref2.searchType;
  return composeSearchEvent({
    event: EVENTS.BROADEN_SEARCH,
    gtmAction: ACTIONS.BROADEN,
    gtmLabel: query,
    searchType: searchType
  });
}

function composeEmptySearchResultsEvent(_ref3) {
  var query = _ref3.query,
      searchType = _ref3.searchType,
      didYouMeanQuerySuggestion = _ref3.didYouMeanQuerySuggestion;
  return composeSearchEvent({
    gtmCategory: CATEGORIES.EMPTY_SEARCH_RESULTS,
    event: EVENTS.EMPTY_SEARCH_RESULT,
    gtmAction: didYouMeanQuerySuggestion ? ACTIONS.RESULT_DID_YOU_MEAN : ACTIONS.RESULT,
    gtmLabel: didYouMeanQuerySuggestion ? didYouMeanQuerySuggestion + " | " + query : query,
    searchType: searchType
  });
}

function composeSaveSearchEvent() {
  return composeSearchEvent({
    event: EVENTS.SAVE_SEARCH,
    gtmCategory: CATEGORIES.SEARCH,
    gtmAction: ACTIONS.SAVE_SEARCH,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.SEARCH_TITLE
  });
}

function composeSearchResultsEvent(_ref4) {
  var query = _ref4.query,
      searchType = _ref4.searchType;
  return composeSearchEvent({
    event: EVENTS.SEARCH_RESULTS,
    gtmAction: ACTIONS.RESULT,
    gtmLabel: query,
    searchType: searchType
  });
}

function composeDidYouMeanClickEvent(_ref5) {
  var query = _ref5.query,
      suggestion = _ref5.suggestion,
      emptySearch = _ref5.emptySearch;
  return composeSearchEvent({
    event: EVENTS.DYM_SEARCH_RESULT_CLICK,
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: suggestion + " | " + query,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.SEARCH_TITLE
  });
}

function composeModifySearchClickEvent() {
  return composeSearchEvent({
    event: EVENTS.MODIFY_SEARCH_CLICK,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: LABELS.MODIFY_MY_SEARCH,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.SEARCH_TITLE
  });
}

function composeExternalSearchClickEvent(_ref6) {
  var emptySearch = _ref6.emptySearch,
      outboundURL = _ref6.outboundURL;
  return composeSearchEvent({
    event: EVENTS.EXTERNAL_SEARCH_CLICK,
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH,
    gtmAction: ACTIONS.OUTBOUND_LINK_CLICK,
    gtmLabel: outboundURL,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.SEARCH_TITLE
  });
}

function composeExternalSearchPanelClickEvent(_ref7) {
  var outboundURL = _ref7.outboundURL,
      linkText = _ref7.linkText;
  return composeSearchEvent({
    event: EVENTS.EXTERNAL_SEARCH_CLICK,
    gtmCategory: CATEGORIES.TRANSACT,
    gtmAction: ACTIONS.OUTBOUND_LINK_CLICK,
    gtmLabel: outboundURL,
    gtmValue: 1,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.EXTERNAL_SEARCH,
    gtmContentCreator: linkText
  });
}

function composeOtherResourcesClickEvent(_ref8) {
  var emptySearch = _ref8.emptySearch,
      label = _ref8.label,
      outboundURL = _ref8.outboundURL;
  return (0, _compose.composeAnalytics)([(0, _compose.composeLinkEvent)(EVENTS.SEARCH_OTHER_RESOURCES_CLICK, outboundURL, {
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH,
    gtmValue: 1,
    gtmLabel: label,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.SEARCH_TITLE,
    gtmContentType: CUSTOM_DIMENSIONS.CONTENT_TYPES.ONLINE_RESOURCE
  })]);
}

function composeDidNotFindPanelToggleEvent(_ref9) {
  var expanded = _ref9.expanded;
  return composeSearchEvent({
    event: EVENTS.DID_NOT_FIND_PANEL_TOGGLE,
    gtmCategory: CATEGORIES.SEARCH_DID_NOT_FIND,
    gtmAction: ACTIONS.INTERACT_WITH_CONTAINER,
    gtmLabel: expanded ? LABELS.CONTAINER_EXPAND : LABELS.CONTAINER_COLLAPSE,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.DID_NOT_FIND_PANEL
  });
}

function composeChatWithLibrarianClickEvent(emptySearch) {
  return composeSearchEvent({
    event: EVENTS.CHAT_WITH_LIBRARIAN_CLICK,
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH_DID_NOT_FIND,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: LABELS.CHAT_WITH_LIBRARIAN,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.DID_NOT_FIND_PANEL
  });
}

function composeDidNotFindResourceClickEvent(_ref10) {
  var text = _ref10.text,
      url = _ref10.url,
      emptySearch = _ref10.emptySearch;
  return (0, _compose.composeAnalytics)([(0, _compose.composeLinkEvent)(EVENTS.DID_NOT_FIND_RESOURCE_CLICK, url, {
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH_DID_NOT_FIND,
    gtmValue: 1,
    gtmLabel: text,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.DID_NOT_FIND_PANEL_RESOURCES
  })]);
}

function composeDidNotFindServiceClickEvent(_ref11) {
  var text = _ref11.text,
      url = _ref11.url,
      emptySearch = _ref11.emptySearch;
  return (0, _compose.composeAnalytics)([(0, _compose.composeLinkEvent)(EVENTS.DID_NOT_FIND_SERVICE_CLICK, url, {
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH_DID_NOT_FIND,
    gtmValue: 1,
    gtmLabel: text,
    gtmContainerName: CUSTOM_DIMENSIONS.CONTAINER_NAMES.DID_NOT_FIND_PANEL_SERVICES
  })]);
}

function composeDidNotFindMoreResourcesClickEvent(_ref12) {
  var url = _ref12.url,
      emptySearch = _ref12.emptySearch;
  return (0, _compose.composeAnalytics)([(0, _compose.composeLinkEvent)(EVENTS.DID_NOT_FIND_MORE_RESOURCES_CLICK, url, {
    gtmCategory: emptySearch ? CATEGORIES.EMPTY_SEARCH_RESULTS : CATEGORIES.SEARCH_DID_NOT_FIND,
    gtmValue: 1,
    gtmLabel: LABELS.VIEW_MORE_RESOURCES
  })]);
}

function composeSearchProbeClickEvent(_ref13) {
  var libraryName = _ref13.libraryName,
      query = _ref13.query;
  return composeSearchEvent({
    event: EVENTS.COMBINED_CATALOG_CLICK,
    gtmCategory: CATEGORIES.SEARCH_COMBINED_CATALOG,
    gtmAction: ACTIONS.CLICK_LINK,
    gtmLabel: [libraryName, query].join(' | '),
    gtmValue: 1
  });
}

function composeInitialPageviewSearchEvent(_ref14) {
  var query = _ref14.query,
      searchType = _ref14.searchType,
      didYouMeanQuerySuggestion = _ref14.didYouMeanQuerySuggestion,
      zeroResults = _ref14.zeroResults;

  if (zeroResults) {
    return composeEmptySearchResultsEvent({
      query: query,
      searchType: searchType,
      didYouMeanQuerySuggestion: didYouMeanQuerySuggestion
    });
  }

  return composeSearchResultsEvent({
    query: query,
    searchType: searchType
  });
}

var _default = {
  composeBroadenSearchEvent: composeBroadenSearchEvent,
  composeEmptySearchResultsEvent: composeEmptySearchResultsEvent,
  composeSaveSearchEvent: composeSaveSearchEvent,
  composeSearchResultsEvent: composeSearchResultsEvent,
  composeDidYouMeanClickEvent: composeDidYouMeanClickEvent,
  composeModifySearchClickEvent: composeModifySearchClickEvent,
  composeExternalSearchClickEvent: composeExternalSearchClickEvent,
  composeOtherResourcesClickEvent: composeOtherResourcesClickEvent,
  composeDidNotFindPanelToggleEvent: composeDidNotFindPanelToggleEvent,
  composeChatWithLibrarianClickEvent: composeChatWithLibrarianClickEvent,
  composeDidNotFindResourceClickEvent: composeDidNotFindResourceClickEvent,
  composeDidNotFindServiceClickEvent: composeDidNotFindServiceClickEvent,
  composeDidNotFindMoreResourcesClickEvent: composeDidNotFindMoreResourcesClickEvent,
  composeSearchProbeClickEvent: composeSearchProbeClickEvent,
  composeInitialPageviewSearchEvent: composeInitialPageviewSearchEvent,
  composeExternalSearchPanelClickEvent: composeExternalSearchPanelClickEvent
};
exports.default = _default;
