// @bibliocommons/analytics-event-composers/lib/similarTitlesEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SIMILAR_TITLES = exports.REMOVE_SIMILAR_TITLE = exports.REMOVE_CONTENT = exports.ADD_SIMILAR_TITLE = exports.ADD_CONTENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var SIMILAR_TITLES = _constantsAnalytics.default.CATEGORIES.SIMILAR_TITLES;
exports.SIMILAR_TITLES = SIMILAR_TITLES;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_SIMILAR_TITLE = _analyticsConstants$E.ADD_SIMILAR_TITLE,
    REMOVE_SIMILAR_TITLE = _analyticsConstants$E.REMOVE_SIMILAR_TITLE;
exports.REMOVE_SIMILAR_TITLE = REMOVE_SIMILAR_TITLE;
exports.ADD_SIMILAR_TITLE = ADD_SIMILAR_TITLE;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.ADD_CONTENT = ADD_CONTENT;

function composeSimilarTitlesEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: SIMILAR_TITLES,
    gtmAction: action
  });
}

function composeAddSimilarTitlesEvent() {
  return (0, _compose.composeAnalytics)([composeSimilarTitlesEvent(ADD_SIMILAR_TITLE, ADD_CONTENT)]);
}

function composeRemoveSimilarTitlesEvent() {
  return (0, _compose.composeAnalytics)([composeSimilarTitlesEvent(REMOVE_SIMILAR_TITLE, REMOVE_CONTENT)]);
}

var _default = {
  composeAddSimilarTitlesEvent: composeAddSimilarTitlesEvent,
  composeRemoveSimilarTitlesEvent: composeRemoveSimilarTitlesEvent
};
exports.default = _default;
