// @bibliocommons/analytics-event-composers/lib/summariesEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var SUMMARY = _constantsAnalytics.default.CATEGORIES.SUMMARY;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_SUMMARY = _analyticsConstants$E.ADD_SUMMARY,
    EDIT_SUMMARY = _analyticsConstants$E.EDIT_SUMMARY,
    REMOVE_SUMMARY = _analyticsConstants$E.REMOVE_SUMMARY;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;

function composeSummaryEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: SUMMARY,
    gtmAction: action
  });
}

function composeAddSummaryEvent() {
  return (0, _compose.composeAnalytics)([composeSummaryEvent(ADD_SUMMARY, ADD_CONTENT)]);
}

function composeEditSummaryEvent() {
  return (0, _compose.composeAnalytics)([composeSummaryEvent(EDIT_SUMMARY, EDIT_ITEM)]);
}

function composeRemoveSummaryEvent() {
  return (0, _compose.composeAnalytics)([composeSummaryEvent(REMOVE_SUMMARY, REMOVE_CONTENT)]);
}

var _default = {
  composeAddSummaryEvent: composeAddSummaryEvent,
  composeEditSummaryEvent: composeEditSummaryEvent,
  composeRemoveSummaryEvent: composeRemoveSummaryEvent
};
exports.default = _default;
