// @bibliocommons/analytics-event-composers/lib/videosEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.VIDEOS = exports.REMOVE_VIDEO = exports.REMOVE_CONTENT = exports.EDIT_VIDEO = exports.EDIT_ITEM = exports.ADD_VIDEO = exports.ADD_CONTENT = void 0;

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var VIDEOS = _constantsAnalytics.default.CATEGORIES.VIDEOS;
exports.VIDEOS = VIDEOS;
var _analyticsConstants$E = _constantsAnalytics.default.EVENTS,
    ADD_VIDEO = _analyticsConstants$E.ADD_VIDEO,
    EDIT_VIDEO = _analyticsConstants$E.EDIT_VIDEO,
    REMOVE_VIDEO = _analyticsConstants$E.REMOVE_VIDEO;
exports.REMOVE_VIDEO = REMOVE_VIDEO;
exports.EDIT_VIDEO = EDIT_VIDEO;
exports.ADD_VIDEO = ADD_VIDEO;
var _analyticsConstants$A = _constantsAnalytics.default.ACTIONS,
    ADD_CONTENT = _analyticsConstants$A.ADD_CONTENT,
    EDIT_ITEM = _analyticsConstants$A.EDIT_ITEM,
    REMOVE_CONTENT = _analyticsConstants$A.REMOVE_CONTENT;
exports.REMOVE_CONTENT = REMOVE_CONTENT;
exports.EDIT_ITEM = EDIT_ITEM;
exports.ADD_CONTENT = ADD_CONTENT;

function composeRatingEvent(eventName, action) {
  return (0, _compose.composeEvent)(eventName, {
    gtmValue: 1,
    gtmLabel: undefined,
    gtmCategory: VIDEOS,
    gtmAction: action
  });
}

function composeAddVideoEvent() {
  return (0, _compose.composeAnalytics)([composeRatingEvent(ADD_VIDEO, ADD_CONTENT)]);
}

function composeEditVideoEvent() {
  return (0, _compose.composeAnalytics)([composeRatingEvent(EDIT_VIDEO, EDIT_ITEM)]);
}

function composeRemoveVideoEvent() {
  return (0, _compose.composeAnalytics)([composeRatingEvent(REMOVE_VIDEO, REMOVE_CONTENT)]);
}

var _default = {
  composeAddVideoEvent: composeAddVideoEvent,
  composeEditVideoEvent: composeEditVideoEvent,
  composeRemoveVideoEvent: composeRemoveVideoEvent
};
exports.default = _default;
