// @bibliocommons/analytics-event-composers/lib/whileYouWaitEventComposer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.composeCoverClickEvent = composeCoverClickEvent;
exports.composeToggleWidgetEvent = composeToggleWidgetEvent;
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _constantsAnalytics = _interopRequireDefault(require("@bibliocommons/constants-analytics"));

var _compose = require("./compose");

var _excluded = ["event", "gtmAction", "gtmLabel", "gtmValue", "metadataIds"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var EVENTS = _constantsAnalytics.default.EVENTS,
    CATEGORIES = _constantsAnalytics.default.CATEGORIES,
    ACTIONS = _constantsAnalytics.default.ACTIONS,
    LABELS = _constantsAnalytics.default.LABELS,
    CUSTOM_DIMENSIONS = _constantsAnalytics.default.CUSTOM_DIMENSIONS;

function composeWhileYouWaitEvent(_ref) {
  var event = _ref.event,
      gtmAction = _ref.gtmAction,
      gtmLabel = _ref.gtmLabel,
      _ref$gtmValue = _ref.gtmValue,
      gtmValue = _ref$gtmValue === void 0 ? 1 : _ref$gtmValue,
      metadataIds = _ref.metadataIds,
      payload = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  return (0, _compose.composeAnalytics)([(0, _compose.composeEvent)(event, _objectSpread({
    gtmCategory: CATEGORIES.WHILE_YOU_WAIT,
    gtmAction: gtmAction,
    gtmLabel: gtmLabel,
    gtmValue: gtmValue,
    gtmContentType: CUSTOM_DIMENSIONS.CONTENT_TYPES.BIB,
    metadataIds: metadataIds
  }, payload))]);
}

function composeToggleWidgetEvent(_ref2) {
  var open = _ref2.open;
  return composeWhileYouWaitEvent({
    event: EVENTS.WHILE_YOU_WAIT_TOGGLE_WIDGET,
    gtmAction: ACTIONS.TOGGLE_OPTION,
    gtmLabel: open ? LABELS.OPEN : LABELS.CLOSE
  });
}

function composeCoverClickEvent(_ref3) {
  var metadataId = _ref3.metadataId;
  return composeWhileYouWaitEvent({
    event: EVENTS.WHILE_YOU_WAIT_COVER_CLICK,
    gtmAction: ACTIONS.CLICK_IMAGE,
    gtmLabel: LABELS.BIB_COVER,
    metadataIds: [metadataId]
  });
}

var _default = {
  composeToggleWidgetEvent: composeToggleWidgetEvent,
  composeCoverClickEvent: composeCoverClickEvent
};
exports.default = _default;
