// @bibliocommons/base-dropdown/lib/components/dropdown-button/components/DropdownButton.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = DropdownButton;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _baseButtons = require("@bibliocommons/base-buttons");

var _bcPropTypes = require("@bibliocommons/bc-prop-types");

require("../styles/DropdownButton.scss");

var _excluded = ["className", "children", "block", "rounded"];

function DropdownButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      block = _ref.block,
      rounded = _ref.rounded,
      props = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  var classNames = (0, _classnames.default)('cp-dropdown-button', className, {
    'dropdown-button--rounded': rounded,
    'dropdown-button--block': block
  });
  return /*#__PURE__*/_react.default.createElement(_baseButtons.Button, (0, _extends2.default)({
    className: classNames,
    dataKey: "dropdown-button"
  }, props), children);
}

DropdownButton.propTypes = {
  children: _bcPropTypes.childrenShape,
  className: _propTypes.default.string,
  isLoading: _propTypes.default.bool,
  block: _propTypes.default.bool,
  rounded: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  handleClick: _propTypes.default.func
};
