// @bibliocommons/bc-prop-types/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "accountShape", {
  enumerable: true,
  get: function get() {
    return _accountShape.default;
  }
});
Object.defineProperty(exports, "addTitleResultShape", {
  enumerable: true,
  get: function get() {
    return _addTitleResultShape.default;
  }
});
Object.defineProperty(exports, "availabilityShape", {
  enumerable: true,
  get: function get() {
    return _availabilityShape.default;
  }
});
Object.defineProperty(exports, "bibBriefShape", {
  enumerable: true,
  get: function get() {
    return _bibBriefShape.default;
  }
});
Object.defineProperty(exports, "bibFieldShape", {
  enumerable: true,
  get: function get() {
    return _bibFieldShape.default;
  }
});
Object.defineProperty(exports, "bibShape", {
  enumerable: true,
  get: function get() {
    return _bibShape.default;
  }
});
Object.defineProperty(exports, "borrowingHistoryItemShape", {
  enumerable: true,
  get: function get() {
    return _borrowingHistoryItemShape.default;
  }
});
Object.defineProperty(exports, "borrowingQuotaShape", {
  enumerable: true,
  get: function get() {
    return _borrowingQuotaShape.default;
  }
});
Object.defineProperty(exports, "branchShape", {
  enumerable: true,
  get: function get() {
    return _branchShape.default;
  }
});
Object.defineProperty(exports, "cardShape", {
  enumerable: true,
  get: function get() {
    return _cardShape.default;
  }
});
Object.defineProperty(exports, "carouselSettingShape", {
  enumerable: true,
  get: function get() {
    return _CarouselSettingShape.default;
  }
});
Object.defineProperty(exports, "catalogBibShape", {
  enumerable: true,
  get: function get() {
    return _catalogBibShape.default;
  }
});
Object.defineProperty(exports, "checkedOutItemShape", {
  enumerable: true,
  get: function get() {
    return _checkedOutItemShape.default;
  }
});
Object.defineProperty(exports, "childrenShape", {
  enumerable: true,
  get: function get() {
    return _childrenShape.default;
  }
});
Object.defineProperty(exports, "deprecatedDropdownOptionShape", {
  enumerable: true,
  get: function get() {
    return _deprecatedDropdownOptionShape.default;
  }
});
Object.defineProperty(exports, "didNotFindOptionsShape", {
  enumerable: true,
  get: function get() {
    return _didNotFindOptionsShape.default;
  }
});
Object.defineProperty(exports, "dropdownOptionShape", {
  enumerable: true,
  get: function get() {
    return _dropdownOptionShape.default;
  }
});
Object.defineProperty(exports, "featuredContentShape", {
  enumerable: true,
  get: function get() {
    return _featuredContentShape.default;
  }
});
Object.defineProperty(exports, "fieldErrorsShape", {
  enumerable: true,
  get: function get() {
    return _fieldErrorsShape.default;
  }
});
Object.defineProperty(exports, "filterShape", {
  enumerable: true,
  get: function get() {
    return _filterShape.default;
  }
});
Object.defineProperty(exports, "holdItemShape", {
  enumerable: true,
  get: function get() {
    return _holdItemShape.default;
  }
});
Object.defineProperty(exports, "holdingShape", {
  enumerable: true,
  get: function get() {
    return _holdingShape.default;
  }
});
Object.defineProperty(exports, "iconShape", {
  enumerable: true,
  get: function get() {
    return _IconShape.default;
  }
});
Object.defineProperty(exports, "jacketShape", {
  enumerable: true,
  get: function get() {
    return _jacketShape.default;
  }
});
Object.defineProperty(exports, "libraryShape", {
  enumerable: true,
  get: function get() {
    return _libraryShape.default;
  }
});
Object.defineProperty(exports, "listItemShape", {
  enumerable: true,
  get: function get() {
    return _listItemShape.default;
  }
});
Object.defineProperty(exports, "listShape", {
  enumerable: true,
  get: function get() {
    return _listShape.default;
  }
});
Object.defineProperty(exports, "marcFieldShape", {
  enumerable: true,
  get: function get() {
    return _marcFieldShape.default;
  }
});
Object.defineProperty(exports, "messagingShape", {
  enumerable: true,
  get: function get() {
    return _messagingShape.default;
  }
});
Object.defineProperty(exports, "onlineResourceShape", {
  enumerable: true,
  get: function get() {
    return _onlineResourceShape.default;
  }
});
Object.defineProperty(exports, "paginationShape", {
  enumerable: true,
  get: function get() {
    return _paginationShape.default;
  }
});
Object.defineProperty(exports, "promoShape", {
  enumerable: true,
  get: function get() {
    return _promoShape.default;
  }
});
Object.defineProperty(exports, "recommendationShape", {
  enumerable: true,
  get: function get() {
    return _recommendationShape.default;
  }
});
Object.defineProperty(exports, "reviewShape", {
  enumerable: true,
  get: function get() {
    return _reviewShape.default;
  }
});
Object.defineProperty(exports, "routerShape", {
  enumerable: true,
  get: function get() {
    return _routerShape.default;
  }
});
Object.defineProperty(exports, "searchFieldShape", {
  enumerable: true,
  get: function get() {
    return _searchFieldShape.default;
  }
});
Object.defineProperty(exports, "searchQueryShape", {
  enumerable: true,
  get: function get() {
    return _searchQueryShape.default;
  }
});
Object.defineProperty(exports, "shelfItemShape", {
  enumerable: true,
  get: function get() {
    return _shelfItemShape.default;
  }
});
Object.defineProperty(exports, "shelfShape", {
  enumerable: true,
  get: function get() {
    return _shelfShape.default;
  }
});
Object.defineProperty(exports, "singleClickHoldsSettingsShape", {
  enumerable: true,
  get: function get() {
    return _singleClickHoldsSettingsShape.default;
  }
});
Object.defineProperty(exports, "siteShape", {
  enumerable: true,
  get: function get() {
    return _siteShape.default;
  }
});
Object.defineProperty(exports, "stackMapShape", {
  enumerable: true,
  get: function get() {
    return _stackMapShape.default;
  }
});
Object.defineProperty(exports, "subscriptionShape", {
  enumerable: true,
  get: function get() {
    return _subscriptionShape.default;
  }
});
Object.defineProperty(exports, "truncateOptionsShape", {
  enumerable: true,
  get: function get() {
    return _truncateOptionsShape.default;
  }
});
Object.defineProperty(exports, "ugcAgeSuitabilityShape", {
  enumerable: true,
  get: function get() {
    return _ugcAgeSuitabilityShape.default;
  }
});
Object.defineProperty(exports, "ugcCommentShape", {
  enumerable: true,
  get: function get() {
    return _ugcCommentShape.default;
  }
});
Object.defineProperty(exports, "ugcContentAdvisoryShape", {
  enumerable: true,
  get: function get() {
    return _ugcContentAdvisoryShape.default;
  }
});
Object.defineProperty(exports, "ugcFlagShape", {
  enumerable: true,
  get: function get() {
    return _ugcFlagShape.default;
  }
});
Object.defineProperty(exports, "ugcIOwnThisShape", {
  enumerable: true,
  get: function get() {
    return _ugcIOwnThisShape.default;
  }
});
Object.defineProperty(exports, "ugcPrivateNoteShape", {
  enumerable: true,
  get: function get() {
    return _ugcPrivateNoteShape.default;
  }
});
Object.defineProperty(exports, "ugcQuotationShape", {
  enumerable: true,
  get: function get() {
    return _ugcQuotationShape.default;
  }
});
Object.defineProperty(exports, "ugcRatingShape", {
  enumerable: true,
  get: function get() {
    return _ugcRatingShape.default;
  }
});
Object.defineProperty(exports, "ugcSimilarTitlesShape", {
  enumerable: true,
  get: function get() {
    return _ugcSimilarTitlesShape.default;
  }
});
Object.defineProperty(exports, "ugcSummaryShape", {
  enumerable: true,
  get: function get() {
    return _ugcSummaryShape.default;
  }
});
Object.defineProperty(exports, "ugcTagShape", {
  enumerable: true,
  get: function get() {
    return _ugcTagShape.default;
  }
});
Object.defineProperty(exports, "ugcVideoShape", {
  enumerable: true,
  get: function get() {
    return _ugcVideoShape.default;
  }
});
Object.defineProperty(exports, "userIdentityShape", {
  enumerable: true,
  get: function get() {
    return _userIdentityShape.default;
  }
});

var _filterShape = _interopRequireDefault(require("./components/filterShape"));

var _searchFieldShape = _interopRequireDefault(require("./components/searchFieldShape"));

var _shelfItemShape = _interopRequireDefault(require("./components/shelfItemShape"));

var _listShape = _interopRequireDefault(require("./components/listShape"));

var _listItemShape = _interopRequireDefault(require("./components/listItemShape"));

var _availabilityShape = _interopRequireDefault(require("./components/availabilityShape"));

var _featuredContentShape = _interopRequireDefault(require("./components/featuredContentShape"));

var _shelfShape = _interopRequireDefault(require("./components/shelfShape"));

var _userIdentityShape = _interopRequireDefault(require("./components/userIdentityShape"));

var _bibBriefShape = _interopRequireDefault(require("./components/bibBriefShape"));

var _onlineResourceShape = _interopRequireDefault(require("./components/onlineResourceShape"));

var _holdingShape = _interopRequireDefault(require("./components/holdingShape"));

var _bibShape = _interopRequireDefault(require("./components/bibShape"));

var _cardShape = _interopRequireDefault(require("./components/cardShape"));

var _catalogBibShape = _interopRequireDefault(require("./components/catalogBibShape"));

var _checkedOutItemShape = _interopRequireDefault(require("./components/checkedOutItemShape"));

var _borrowingQuotaShape = _interopRequireDefault(require("./components/borrowingQuotaShape"));

var _holdItemShape = _interopRequireDefault(require("./components/holdItemShape"));

var _branchShape = _interopRequireDefault(require("./components/branchShape"));

var _singleClickHoldsSettingsShape = _interopRequireDefault(require("./components/singleClickHoldsSettingsShape"));

var _IconShape = _interopRequireDefault(require("./components/IconShape"));

var _CarouselSettingShape = _interopRequireDefault(require("./components/CarouselSettingShape"));

var _childrenShape = _interopRequireDefault(require("./components/childrenShape"));

var _libraryShape = _interopRequireDefault(require("./components/libraryShape"));

var _siteShape = _interopRequireDefault(require("./components/siteShape"));

var _accountShape = _interopRequireDefault(require("./components/accountShape"));

var _messagingShape = _interopRequireDefault(require("./components/messagingShape"));

var _jacketShape = _interopRequireDefault(require("./components/jacketShape"));

var _recommendationShape = _interopRequireDefault(require("./components/recommendationShape"));

var _ugcCommentShape = _interopRequireDefault(require("./components/ugcCommentShape"));

var _ugcQuotationShape = _interopRequireDefault(require("./components/ugcQuotationShape"));

var _ugcRatingShape = _interopRequireDefault(require("./components/ugcRatingShape"));

var _ugcTagShape = _interopRequireDefault(require("./components/ugcTagShape"));

var _ugcVideoShape = _interopRequireDefault(require("./components/ugcVideoShape"));

var _paginationShape = _interopRequireDefault(require("./components/paginationShape"));

var _fieldErrorsShape = _interopRequireDefault(require("./components/fieldErrorsShape"));

var _ugcIOwnThisShape = _interopRequireDefault(require("./components/ugcIOwnThisShape"));

var _ugcSimilarTitlesShape = _interopRequireDefault(require("./components/ugcSimilarTitlesShape"));

var _ugcPrivateNoteShape = _interopRequireDefault(require("./components/ugcPrivateNoteShape"));

var _addTitleResultShape = _interopRequireDefault(require("./components/addTitleResultShape"));

var _searchQueryShape = _interopRequireDefault(require("./components/searchQueryShape"));

var _truncateOptionsShape = _interopRequireDefault(require("./components/truncateOptionsShape"));

var _ugcContentAdvisoryShape = _interopRequireDefault(require("./components/ugcContentAdvisoryShape"));

var _ugcAgeSuitabilityShape = _interopRequireDefault(require("./components/ugcAgeSuitabilityShape"));

var _ugcSummaryShape = _interopRequireDefault(require("./components/ugcSummaryShape"));

var _borrowingHistoryItemShape = _interopRequireDefault(require("./components/borrowingHistoryItemShape"));

var _didNotFindOptionsShape = _interopRequireDefault(require("./components/didNotFindOptionsShape"));

var _bibFieldShape = _interopRequireDefault(require("./components/bibFieldShape"));

var _ugcFlagShape = _interopRequireDefault(require("./components/ugcFlagShape"));

var _subscriptionShape = _interopRequireDefault(require("./components/subscriptionShape"));

var _stackMapShape = _interopRequireDefault(require("./components/stackMapShape"));

var _routerShape = _interopRequireDefault(require("./components/routerShape"));

var _reviewShape = _interopRequireDefault(require("./components/reviewShape"));

var _promoShape = _interopRequireDefault(require("./components/promoShape"));

var _marcFieldShape = _interopRequireDefault(require("./components/marcFieldShape"));

var _deprecatedDropdownOptionShape = _interopRequireDefault(require("./components/deprecatedDropdownOptionShape"));

var _dropdownOptionShape = _interopRequireDefault(require("./components/dropdownOptionShape"));
