// @bibliocommons/alerts/lib/components/alert/components/Alert.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _reactIntl = require("react-intl");

var _classnames = _interopRequireDefault(require("classnames"));

var _baseScreenReaderMessage = _interopRequireDefault(require("@bibliocommons/base-screen-reader-message"));

var _bcPropTypes = require("@bibliocommons/bc-prop-types");

var _baseButtons = require("@bibliocommons/base-buttons");

var _baseIcons = require("@bibliocommons/base-icons");

require("../styles/Alert.scss");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var SCREEN_READER_CONTENT_TIMEOUT = 100;

var Alert = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(Alert, _React$PureComponent);

  var _super = _createSuper(Alert);

  function Alert(props) {
    var _this;

    (0, _classCallCheck2.default)(this, Alert);
    _this = _super.call(this, props);
    _this.state = {
      showScreenReaderContent: false
    };
    return _this;
  }

  (0, _createClass2.default)(Alert, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this2 = this;

      // NVDA only announces live regions which are added to the DOM directly.
      // For example, it does not work if a div _containing_ an aria-live region is added.
      // Adding the live region after a timeout ensures it's added separately from the element that contains it.
      this._intervalId = setTimeout(function () {
        return _this2.setState({
          showScreenReaderContent: true
        });
      }, SCREEN_READER_CONTENT_TIMEOUT);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      clearInterval(this._intervalId);
    }
  }, {
    key: "renderIcon",
    value: function renderIcon() {
      var renderIcon = this.props.renderIcon;
      return renderIcon && /*#__PURE__*/_react.default.createElement("span", {
        className: "alert__icon"
      }, renderIcon());
    }
  }, {
    key: "renderDismissButton",
    value: function renderDismissButton() {
      var handleDismiss = this.props.handleDismiss;
      return handleDismiss && /*#__PURE__*/_react.default.createElement(_baseButtons.IconButton, {
        className: "alert__dismiss",
        "data-key": "alert-dismiss",
        handleClick: handleDismiss
      }, /*#__PURE__*/_react.default.createElement(_baseIcons.Close, null), /*#__PURE__*/_react.default.createElement(_baseScreenReaderMessage.default, null, /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
        id: "dismiss_alert"
      })));
    }
  }, {
    key: "renderContent",
    value: function renderContent() {
      var children = this.props.children;
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "alert__content"
      }, /*#__PURE__*/_react.default.createElement("div", {
        "aria-hidden": true
      }, children), this.state.showScreenReaderContent && /*#__PURE__*/_react.default.createElement("div", {
        role: "alert",
        "aria-live": "polite",
        "aria-atomic": "true"
      }, /*#__PURE__*/_react.default.createElement(_baseScreenReaderMessage.default, null, children)));
    }
  }, {
    key: "render",
    value: function render() {
      var classNames = (0, _classnames.default)('cp-alert', this.props.className);
      return /*#__PURE__*/_react.default.createElement("div", {
        className: classNames,
        tabIndex: "-1"
      }, this.renderIcon(), this.renderContent(), this.renderDismissButton());
    }
  }]);
  return Alert;
}(_react.default.PureComponent);

exports.default = Alert;
Alert.propTypes = {
  className: _propTypes.default.string,
  renderIcon: _propTypes.default.func,
  handleDismiss: _propTypes.default.func,
  children: _bcPropTypes.childrenShape.isRequired
};
