// @bibliocommons/base-links/lib/components/Link.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _url = _interopRequireDefault(require("url"));

var _reactIntl = require("react-intl");

var _utilsReact = require("@bibliocommons/utils-react");

var _baseScreenReaderMessage = _interopRequireDefault(require("@bibliocommons/base-screen-reader-message"));

var _utilsHtml = require("@bibliocommons/utils-html");

var _context = _interopRequireDefault(require("@bibliocommons/context"));

require("../styles/Links.scss");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var Link = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(Link, _React$PureComponent);

  var _super = _createSuper(Link);

  function Link(props) {
    var _this;

    (0, _classCallCheck2.default)(this, Link);
    _this = _super.call(this, props);
    (0, _utilsReact.bindAll)((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(Link, [{
    key: "handleClick",
    value: function handleClick(e) {
      var _this$props = this.props,
          preventDefault = _this$props.preventDefault,
          handleClick = _this$props.handleClick;

      if (preventDefault) {
        e.preventDefault();
      }

      if (handleClick) {
        handleClick(e);
      }
    }
  }, {
    key: "renderContent",
    value: function renderContent() {
      var _this$props2 = this.props,
          target = _this$props2.target,
          screenReaderText = _this$props2.screenReaderText,
          children = _this$props2.children;

      if (target === '_blank') {
        return /*#__PURE__*/_react.default.createElement("span", null, /*#__PURE__*/_react.default.createElement("span", {
          "aria-hidden": true
        }, children), /*#__PURE__*/_react.default.createElement(_baseScreenReaderMessage.default, null, /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
          id: "ada_link_opens_new_window",
          values: {
            linkText: screenReaderText || children
          }
        })));
      }

      return children;
    }
  }, {
    key: "render",
    value: function render() {
      var _cn;

      var _this$props3 = this.props,
          href = _this$props3.href,
          className = _this$props3.className,
          block = _this$props3.block,
          rounded = _this$props3.rounded,
          size = _this$props3.size,
          dataKey = _this$props3.dataKey,
          target = _this$props3.target;
      var classNames = (0, _classnames.default)('cp-link', className, (_cn = {
        'link--block': block,
        'link--rounded': rounded
      }, _cn["link--" + size] = !!size, _cn));
      var baseURL = this.context && this.context.baseURL;

      var linkProps = _objectSpread(_objectSpread({}, (0, _utilsHtml.getValidHtmlAttributes)(this.props, 'a')), {}, {
        className: classNames,
        onClick: this.handleClick,
        'data-key': dataKey,
        href: baseURL ? _url.default.resolve(baseURL, href) : href
      }); // jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/


      if (target === '_blank') {
        linkProps.rel = 'noopener noreferrer';
      }

      return /*#__PURE__*/_react.default.createElement("a", linkProps, this.renderContent());
    }
  }]);
  return Link;
}(_react.default.PureComponent);

exports.default = Link;
Link.contextType = _context.default;
Link.propTypes = {
  children: _propTypes.default.oneOfType([_propTypes.default.arrayOf(_propTypes.default.node), _propTypes.default.node]).isRequired,
  className: _propTypes.default.string,
  target: _propTypes.default.string,
  screenReaderText: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.element]),
  dataKey: _propTypes.default.string.isRequired,
  href: _propTypes.default.string.isRequired,
  preventDefault: _propTypes.default.bool,
  handleClick: _propTypes.default.func,
  block: _propTypes.default.bool,
  rounded: _propTypes.default.bool,
  size: _propTypes.default.oneOf(['default', 'small'], 'default')
};
