// @bibliocommons/modal/lib/components/modal-content/components/ModalContent.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _bcPropTypes = require("@bibliocommons/bc-prop-types");

var _throttle = _interopRequireDefault(require("lodash/throttle"));

var _classnames = _interopRequireDefault(require("classnames"));

require("../styles/ModalContent.scss");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var ModalContent = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(ModalContent, _React$PureComponent);

  var _super = _createSuper(ModalContent);

  function ModalContent(props) {
    var _this;

    (0, _classCallCheck2.default)(this, ModalContent);
    _this = _super.call(this, props);
    _this.contentRef = null;
    _this.handleScroll = (0, _throttle.default)(_this.handleScroll, 50, {
      leading: true,
      trailing: true
    }).bind((0, _assertThisInitialized2.default)(_this));
    _this.state = {
      scrolled: false
    };
    return _this;
  }

  (0, _createClass2.default)(ModalContent, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (this.contentRef) {
        this.contentRef.addEventListener('scroll', this.handleScroll);
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (this.contentRef) {
        this.contentRef.removeEventListener('scroll', this.handleScroll);
      }
    }
  }, {
    key: "handleScroll",
    value: function handleScroll() {
      if (this.contentRef) {
        this.setState({
          scrolled: this.contentRef.scrollTop > 1
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children;
      var classNames = (0, _classnames.default)('cp-modal-content', className, {
        'modal-content--scrolled': this.state.scrolled
      });
      return /*#__PURE__*/_react.default.createElement("div", {
        className: classNames,
        ref: function ref(el) {
          return _this2.contentRef = el;
        }
      }, children);
    }
  }]);
  return ModalContent;
}(_react.default.PureComponent);

exports.default = ModalContent;
ModalContent.propTypes = {
  className: _propTypes.default.string,
  children: _bcPropTypes.childrenShape
};
