// @bibliocommons/rating/lib/components/rating-stars/components/RatingStars.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RatingStars = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _times = _interopRequireDefault(require("lodash/times"));

var _uniqueId = _interopRequireDefault(require("lodash/uniqueId"));

var _utilsReact = require("@bibliocommons/utils-react");

var _baseScreenReaderMessage = _interopRequireDefault(require("@bibliocommons/base-screen-reader-message"));

var _baseIcons = require("@bibliocommons/base-icons");

require("../styles/RatingStars.scss");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var starCount = {
  fullStars: function fullStars(n) {
    return Math.floor(n / 20);
  },
  halfStars: function halfStars(n) {
    return n % 20 > 0 ? 1 : 0;
  },
  emptyStars: function emptyStars(n) {
    return 5 - Math.round(n / 20);
  }
};

var RatingStars = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(RatingStars, _React$PureComponent);

  var _super = _createSuper(RatingStars);

  function RatingStars(props) {
    var _this;

    (0, _classCallCheck2.default)(this, RatingStars);
    _this = _super.call(this, props);
    (0, _utilsReact.bindAll)((0, _assertThisInitialized2.default)(_this));
    _this.id = (0, _uniqueId.default)('rating-');
    return _this;
  }

  (0, _createClass2.default)(RatingStars, [{
    key: "drawStars",
    value: function drawStars() {
      var _this2 = this;

      var _this$props = this.props,
          rating = _this$props.rating,
          size = _this$props.size;
      var stars = [];
      var normalizedRating = Math.round(rating / 10) * 10;
      (0, _times.default)(starCount.fullStars(normalizedRating), function (n) {
        return stars.push( /*#__PURE__*/_react.default.createElement(_baseIcons.Star, {
          key: "star-" + n + "-" + _this2.id,
          className: "rating-stars__icon--" + size
        }));
      });
      (0, _times.default)(starCount.halfStars(normalizedRating), function (n) {
        return stars.push( /*#__PURE__*/_react.default.createElement(_baseIcons.HalfStar, {
          key: "half-star-" + n + "-" + _this2.id,
          className: "rating-stars__icon--" + size
        }));
      });
      (0, _times.default)(starCount.emptyStars(normalizedRating), function (n) {
        return stars.push( /*#__PURE__*/_react.default.createElement(_baseIcons.StarOutline, {
          key: "empty-star-" + n + "-" + _this2.id,
          className: "rating-stars__icon--" + size
        }));
      });
      return stars;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          className = _this$props2.className,
          screenReaderLabel = _this$props2.screenReaderLabel;
      return /*#__PURE__*/_react.default.createElement("span", {
        className: (0, _classnames.default)('cp-rating-stars', className),
        "aria-hidden": !screenReaderLabel
      }, screenReaderLabel && /*#__PURE__*/_react.default.createElement(_baseScreenReaderMessage.default, null, screenReaderLabel), this.drawStars());
    }
  }]);
  return RatingStars;
}(_react.default.PureComponent);

exports.RatingStars = RatingStars;
RatingStars.defaultProps = {
  rating: 0,
  size: 'default'
};
RatingStars.propTypes = {
  className: _propTypes.default.string,
  rating: _propTypes.default.number,
  size: _propTypes.default.oneOf(['default', 'extra-extra-small', 'extra-small', 'small', 'medium', 'large']),
  screenReaderLabel: _propTypes.default.oneOfType([_propTypes.default.node, _propTypes.default.string])
};
var _default = RatingStars;
exports.default = _default;
