// @bibliocommons/heading/lib/components/Heading.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Heading;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _bcPropTypes = require("@bibliocommons/bc-prop-types");

require("../styles/Heading.scss");

var _excluded = ["className", "children", "size", "variant", "linked", "tagName"];

function Heading(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'modest' : _ref$size,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? false : _ref$variant,
      _ref$linked = _ref.linked,
      linked = _ref$linked === void 0 ? true : _ref$linked,
      Tag = _ref.tagName,
      extraProps = (0, _objectWithoutPropertiesLoose2.default)(_ref, _excluded);
  var classNames = (0, _classnames.default)('cp-heading', "heading-" + size, className, {
    'heading--variant': variant,
    'heading--linked': linked
  });
  return /*#__PURE__*/_react.default.createElement(Tag, (0, _extends2.default)({}, extraProps, {
    className: classNames
  }), children);
}

Heading.propTypes = {
  className: _propTypes.default.string,
  children: _bcPropTypes.childrenShape.isRequired,
  tagName: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]).isRequired,
  size: _propTypes.default.oneOf(['giant', 'large', 'moderate', 'medium', 'modest', 'small']),
  variant: _propTypes.default.bool,
  linked: _propTypes.default.bool
};
