// @bibliocommons/rating/lib/components/rating-control/components/RatingControl.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RatingControl = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _immutable = _interopRequireDefault(require("immutable"));

var _reactIntl = require("react-intl");

var _utilsReact = require("@bibliocommons/utils-react");

var _baseButtons = require("@bibliocommons/base-buttons");

var _baseDropdown = require("@bibliocommons/base-dropdown");

var _ratingStars = _interopRequireDefault(require("../../rating-stars"));

require("../styles/RatingControl.scss");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var RatingControl = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(RatingControl, _React$PureComponent);

  var _super = _createSuper(RatingControl);

  function RatingControl(props) {
    var _this;

    (0, _classCallCheck2.default)(this, RatingControl);
    _this = _super.call(this, props);
    (0, _utilsReact.bindAll)((0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(RatingControl, [{
    key: "getOptions",
    value: function getOptions() {
      var _this2 = this;

      var ratingOptions = Array(10).fill(null).map(function (_, rating) {
        var ratingValue = (rating + 1) * 10;
        return _immutable.default.Map({
          key: "rating-dropdown-value-" + (rating + 1),
          selected: _this2.props.rating === ratingValue,
          value: ratingValue,
          label: _this2.props.intl.formatMessage({
            id: 'rating_values'
          }, {
            rating: (rating + 1) * 0.5
          })
        });
      });

      var noRating = _immutable.default.Map({
        key: 'rating-dropdown-no-rating',
        selected: !this.props.rating,
        value: 0,
        label: this.props.intl.formatMessage({
          id: 'no_rating'
        })
      });

      return _immutable.default.List([noRating].concat(ratingOptions));
    }
  }, {
    key: "renderDropdownLabel",
    value: function renderDropdownLabel() {
      var _this$props = this.props,
          rating = _this$props.rating,
          size = _this$props.size;
      var label = rating > 0 ? 'your_rating' : 'rate_this';
      var starsClass = (0, _classnames.default)('rating-control__rating-stars', {
        'rating-control__rating-stars--branded': rating > 0 && this.props.useLibraryBranding,
        'rating-control__rating-stars--generic': rating > 0 && !this.props.useLibraryBranding
      });
      return /*#__PURE__*/_react.default.createElement("span", {
        className: "rating-control__dropdown_label"
      }, /*#__PURE__*/_react.default.createElement(_ratingStars.default, {
        className: starsClass,
        rating: rating,
        size: size,
        screenReaderLabel: /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
          id: label + "_sr_label",
          values: {
            rating: rating / 20
          }
        })
      }), /*#__PURE__*/_react.default.createElement("span", {
        className: "rating-control__label",
        "aria-hidden": true
      }, /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
        id: "" + label
      })));
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: (0, _classnames.default)('cp-rating-control', this.props.className)
      }, /*#__PURE__*/_react.default.createElement(_baseDropdown.Dropdown, {
        selectable: true,
        showLoadingText: true,
        isLoading: this.props.isLoading,
        disabled: this.props.disabled,
        buttonTag: _baseButtons.Button,
        dataKey: "rating-dropdown",
        className: "rating-dropdown",
        buttonClassName: "rating-control__dropdown_button",
        renderLabel: this.renderDropdownLabel,
        options: this.getOptions(),
        handleSelect: this.props.handleSelect
      }));
    }
  }]);
  return RatingControl;
}(_react.default.PureComponent);

exports.RatingControl = RatingControl;
RatingControl.defaultProps = {
  rating: 0,
  size: 'default',
  useLibraryBranding: true
};
RatingControl.propTypes = {
  className: _propTypes.default.string,
  handleSelect: _propTypes.default.func,
  rating: _propTypes.default.number,
  isLoading: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  size: _propTypes.default.oneOf(['default', 'extra-extra-small', 'extra-small', 'small', 'medium', 'large']),
  useLibraryBranding: _propTypes.default.bool
};

var _default = (0, _reactIntl.injectIntl)(RatingControl);

exports.default = _default;
