// @bibliocommons/deprecated-base-alerts/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Alert", {
  enumerable: true,
  get: function get() {
    return _alert.default;
  }
});
Object.defineProperty(exports, "AlertDismiss", {
  enumerable: true,
  get: function get() {
    return _dismiss.default;
  }
});
Object.defineProperty(exports, "AlertFailure", {
  enumerable: true,
  get: function get() {
    return _failure.default;
  }
});
Object.defineProperty(exports, "AlertInfo", {
  enumerable: true,
  get: function get() {
    return _info.default;
  }
});
Object.defineProperty(exports, "AlertNeutral", {
  enumerable: true,
  get: function get() {
    return _neutral.default;
  }
});
Object.defineProperty(exports, "AlertSuccess", {
  enumerable: true,
  get: function get() {
    return _success.default;
  }
});
Object.defineProperty(exports, "AlertWarning", {
  enumerable: true,
  get: function get() {
    return _warning.default;
  }
});

var _alert = _interopRequireDefault(require("./components/alert"));

var _dismiss = _interopRequireDefault(require("./components/dismiss"));

var _failure = _interopRequireDefault(require("./components/failure"));

var _info = _interopRequireDefault(require("./components/info"));

var _success = _interopRequireDefault(require("./components/success"));

var _warning = _interopRequireDefault(require("./components/warning"));

var _neutral = _interopRequireDefault(require("./components/neutral"));
