// @bibliocommons/bc-prop-types/lib/components/accountShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _singleClickHoldsSettingsShape = _interopRequireDefault(require("./singleClickHoldsSettingsShape"));

var accountShape = _reactImmutableProptypes.default.contains({
  id: _propTypes.default.number,
  libId: _propTypes.default.number,
  digitalNotificationEmail: _propTypes.default.string,
  expiryDate: _propTypes.default.string,
  importedILSShelves: _propTypes.default.bool,
  borrowingHistoryEnabled: _propTypes.default.bool,
  singleClickHoldsSettings: _singleClickHoldsSettingsShape.default,
  preferredLocations: _reactImmutableProptypes.default.listOf(_propTypes.default.string)
});

var _default = accountShape;
exports.default = _default;
