// @bibliocommons/bc-prop-types/lib/components/availabilityShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var availabilityShape = _reactImmutableProptypes.default.contains({
  circulationType: _propTypes.default.string,
  localisedStatus: _propTypes.default.string,
  heldCopies: _propTypes.default.number,
  eresourceDescription: _propTypes.default.string,
  singleBranch: _propTypes.default.bool,
  metadataId: _propTypes.default.string,
  availabilityLocationType: _propTypes.default.string,
  status: _propTypes.default.string,
  bibType: _propTypes.default.string,
  totalCopies: _propTypes.default.number,
  libraryUseOnly: _propTypes.default.bool,
  eresourceUrl: _propTypes.default.string,
  availableCopies: _propTypes.default.number
});

var _default = availabilityShape;
exports.default = _default;
