// @bibliocommons/bc-prop-types/lib/components/bibBriefShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.seriesShape = exports.ratingShape = exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _jacketShape = _interopRequireDefault(require("./jacketShape"));

var seriesShape = _reactImmutableProptypes.default.mapContains({
  name: _propTypes.default.string,
  sortName: _propTypes.default.string,
  volume: _propTypes.default.string,
  volumeNumber: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number])
});

exports.seriesShape = seriesShape;

var ratingShape = _reactImmutableProptypes.default.mapContains({
  averageRating: _propTypes.default.number,
  totalCount: _propTypes.default.number
});

exports.ratingShape = ratingShape;

var _default = _reactImmutableProptypes.default.mapContains({
  authors: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  callNumber: _propTypes.default.string,
  consumptionFormat: _propTypes.default.string,
  description: _propTypes.default.string,
  edition: _propTypes.default.string,
  format: _propTypes.default.string,
  id: _propTypes.default.string,
  isbns: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  jacket: _jacketShape.default,
  metadataId: _propTypes.default.string,
  multiscriptAuthor: _propTypes.default.string,
  multiscriptTitle: _propTypes.default.string,
  primaryLanguage: _propTypes.default.string,
  publicationDate: _propTypes.default.string,
  rating: ratingShape,
  series: _reactImmutableProptypes.default.listOf(seriesShape),
  subtitle: _propTypes.default.string,
  superFormats: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  title: _propTypes.default.string
});

exports.default = _default;
