// @bibliocommons/bc-prop-types/lib/components/bibShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.policyShape = exports.materialPolicyShape = exports.default = exports.availabilityShape = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _bibBriefShape = _interopRequireDefault(require("./bibBriefShape"));

var availabilityShape = _reactImmutableProptypes.default.mapContains({
  availabilityLocationType: _propTypes.default.string,
  availableCopies: _propTypes.default.number,
  bibType: _propTypes.default.string,
  circulationType: _propTypes.default.string,
  eresourceDescription: _propTypes.default.string,
  eresourceUrl: _propTypes.default.string,
  heldCopies: _propTypes.default.number,
  libraryUseOnly: _propTypes.default.bool,
  localisedStatus: _propTypes.default.string,
  metadataId: _propTypes.default.string,
  singleBranch: _propTypes.default.bool,
  status: _propTypes.default.string,
  totalCopies: _propTypes.default.number
});

exports.availabilityShape = availabilityShape;

var materialPolicyShape = _reactImmutableProptypes.default.mapContains({
  externalAPI: _propTypes.default.bool,
  healthy: _propTypes.default.bool,
  helpUrl: _propTypes.default.string,
  itemLevelHoldsPredetermined: _propTypes.default.bool,
  quickStartUrl: _propTypes.default.string,
  requiresDuration: _propTypes.default.bool,
  requiresEmailDuringHold: _propTypes.default.bool,
  requiresFormatDuringCheckout: _propTypes.default.bool,
  requiresFormatDuringHold: _propTypes.default.bool,
  showFormatDuringCheckout: _propTypes.default.bool,
  showFormatDuringHold: _propTypes.default.bool,
  showQuickstartLink: _propTypes.default.bool,
  supportAutoCheckout: _propTypes.default.bool,
  supportsBatchVolumeHolds: _propTypes.default.bool,
  supportsDownload: _propTypes.default.bool,
  supportsFastAvailability: _propTypes.default.bool,
  supportsReadNow: _propTypes.default.bool,
  supportsSuspendHold: _propTypes.default.bool
});

exports.materialPolicyShape = materialPolicyShape;

var policyShape = _reactImmutableProptypes.default.mapContains({
  holdable: _propTypes.default.bool,
  itemLevelHoldableItems: _propTypes.default.bool,
  materialPolicy: materialPolicyShape,
  materialType: _propTypes.default.string,
  provider: _propTypes.default.string
});

exports.policyShape = policyShape;

var _default = _reactImmutableProptypes.default.mapContains({
  availability: availabilityShape,
  briefInfo: _bibBriefShape.default,
  id: _propTypes.default.string,
  policy: policyShape,
  sourceLibId: _propTypes.default.number
});

exports.default = _default;
