// @bibliocommons/bc-prop-types/lib/components/cardShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var bwImageShape = _reactImmutableProptypes.default.mapContains({
  isInformative: _propTypes.default.bool,
  altText: _propTypes.default.string,
  url: _propTypes.default.string.isRequired,
  attribution: _reactImmutableProptypes.default.mapContains({
    licenceEnable: _propTypes.default.bool,
    sourceTitle: _propTypes.default.string,
    sourceUrl: _propTypes.default.string,
    sourceAuthor: _propTypes.default.string,
    authorUrl: _propTypes.default.string,
    ccLicenseVersion: _propTypes.default.string,
    ccLicenseUrl: _propTypes.default.string
  })
});

var _default = _reactImmutableProptypes.default.mapContains({
  id: _propTypes.default.number.isRequired,
  contentType: _reactImmutableProptypes.default.mapContains({
    label: _propTypes.default.string,
    slug: _propTypes.default.string
  }),
  title: _reactImmutableProptypes.default.mapContains({
    text: _propTypes.default.string.isRequired
  }).isRequired,
  description: _reactImmutableProptypes.default.mapContains({
    text: _propTypes.default.string
  }),
  url: _propTypes.default.string.isRequired,
  author: _reactImmutableProptypes.default.mapContains({
    url: _propTypes.default.string,
    name: _propTypes.default.string.isRequired
  }),
  image: _reactImmutableProptypes.default.mapContains({
    bwCardImage11: bwImageShape,
    bwCardImage21: bwImageShape
  }),
  event: _reactImmutableProptypes.default.mapContains({
    type: _propTypes.default.oneOf(['EVENT_BIBLIOEVENTS', 'EVENT_MANUAL']).isRequired,
    date: _propTypes.default.string,
    startTime: _propTypes.default.string,
    endTime: _propTypes.default.string
  }),
  location: _reactImmutableProptypes.default.mapContains({
    name: _propTypes.default.string.isRequired,
    url: _propTypes.default.string.isRequired
  })
});

exports.default = _default;
