// @bibliocommons/bc-prop-types/lib/components/catalogBibShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _default = _reactImmutableProptypes.default.mapContains({
  metadataId: _propTypes.default.string,
  brief: _reactImmutableProptypes.default.mapContains({
    title: _propTypes.default.string,
    otherScriptTitle: _propTypes.default.string,
    subTitle: _propTypes.default.string,
    otherScriptSubTitle: _propTypes.default.string,
    creators: _reactImmutableProptypes.default.listOf(_reactImmutableProptypes.default.mapContains({
      fullName: _propTypes.default.string,
      otherScriptFullName: _propTypes.default.string,
      searchType: _propTypes.default.string,
      otherScriptSearchQuery: _propTypes.default.string,
      relationships: _reactImmutableProptypes.default.list
    })),
    format: _propTypes.default.string,
    primaryLanguage: _propTypes.default.string,
    description: _propTypes.default.string,
    publicationDate: _propTypes.default.string,
    edition: _propTypes.default.string,
    otherScriptEdition: _propTypes.default.string,
    coverImage: _reactImmutableProptypes.default.mapContains({
      type: _propTypes.default.oneOf(['LOCAL', 'NONLOCAL']),
      providerURI: _propTypes.default.string,
      small: _propTypes.default.string,
      medium: _propTypes.default.string,
      large: _propTypes.default.string,
      local: _propTypes.default.string
    })
  }),
  fields: _reactImmutableProptypes.default.list
});

exports.default = _default;
