// @bibliocommons/bc-prop-types/lib/components/checkedOutItemShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _bibShape = _interopRequireDefault(require("./bibShape"));

var _default = _reactImmutableProptypes.default.mapContains({
  actions: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  bib: _bibShape.default,
  checkInEnabled: _propTypes.default.bool,
  checkoutId: _propTypes.default.string,
  digitalId: _propTypes.default.string,
  downloadUrl: _propTypes.default.string,
  downloadable: _propTypes.default.bool,
  dueDate: _propTypes.default.string,
  fines: _propTypes.default.number,
  format: _propTypes.default.string,
  materialType: _propTypes.default.string,
  metadataId: _propTypes.default.string,
  status: _propTypes.default.string,
  thirdPartyReaderUrl: _propTypes.default.string,
  timesRenewed: _propTypes.default.number
});

exports.default = _default;
