// @bibliocommons/bc-prop-types/lib/components/didNotFindOptionsShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.didNotFindServicesShape = exports.didNotFindResourcesShape = exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var didNotFindServicesShape = _reactImmutableProptypes.default.mapContains({
  id: _propTypes.default.string,
  localKey: _propTypes.default.string,
  link: _propTypes.default.string
});

exports.didNotFindServicesShape = didNotFindServicesShape;

var didNotFindResourcesShape = _reactImmutableProptypes.default.mapContains({
  resourceName: _propTypes.default.string,
  resourceLink: _propTypes.default.string,
  resourceDescription: _propTypes.default.string,
  resourceTags: _reactImmutableProptypes.default.list
});

exports.didNotFindResourcesShape = didNotFindResourcesShape;

var _default = _reactImmutableProptypes.default.mapContains({
  isEnabled: _propTypes.default.bool,
  didNotFindServices: _reactImmutableProptypes.default.listOf(didNotFindServicesShape),
  didNotFindResources: _reactImmutableProptypes.default.listOf(didNotFindResourcesShape),
  additionalResourcesLink: _propTypes.default.string
});

exports.default = _default;
