// @bibliocommons/bc-prop-types/lib/components/holdItemShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _bibShape = _interopRequireDefault(require("./bibShape"));

var _branchShape = _interopRequireDefault(require("./branchShape"));

var _default = _reactImmutableProptypes.default.mapContains({
  actions: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  bib: _bibShape.default,
  expiryDate: _propTypes.default.string,
  holdPlacedDate: _propTypes.default.string,
  holdsId: _propTypes.default.string,
  holdsPosition: _propTypes.default.number,
  itemId: _propTypes.default.string,
  materialType: _propTypes.default.string,
  metadataId: _propTypes.default.string,
  pickupLocation: _branchShape.default,
  status: _propTypes.default.string
});

exports.default = _default;
