// @bibliocommons/bc-prop-types/lib/components/libraryShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var _siteShape = _interopRequireDefault(require("./siteShape"));

var libraryShape = _reactImmutableProptypes.default.contains({
  getCardLink: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.bool]),
  showHoldCountInSearch: _propTypes.default.bool,
  showDueDateInCirc: _propTypes.default.bool,
  siteId: _propTypes.default.string,
  domain: _propTypes.default.string,
  showCallNumber: _propTypes.default.bool,
  numericPasswordRequired: _propTypes.default.bool,
  shortName: _propTypes.default.string,
  longName: _propTypes.default.string,
  showVolumeSelectorStatus: _propTypes.default.bool,
  singleBranch: _propTypes.default.bool,
  showExternalSearchLink: _propTypes.default.bool,
  logo: _propTypes.default.string,
  featuredContentBlockPosition: _propTypes.default.number,
  showVolumeSelectorDueDate: _propTypes.default.bool,
  showInactiveHolds: _propTypes.default.bool,
  featuredListTypes: _propTypes.default.oneOfType([_reactImmutableProptypes.default.listOf(_propTypes.default.string), _propTypes.default.bool]),
  showCallNumberForDigital: _propTypes.default.bool,
  showVolumeSelectorCollection: _propTypes.default.bool,
  pickupLocationEditable: _propTypes.default.bool,
  libraryId: _propTypes.default.number,
  showCollectionInCirc: _propTypes.default.bool,
  recentlyReturned: _propTypes.default.bool,
  cmsEnabled: _propTypes.default.bool,
  holdInfoMessages: _reactImmutableProptypes.default.list,
  onlyShowEditionsForCDROMAndVideoGames: _propTypes.default.bool,
  fullName: _propTypes.default.string,
  type: _propTypes.default.string,
  id: _propTypes.default.number,
  hideCallNumberForHigherSearchScope: _propTypes.default.bool,
  linkableLibraries: _reactImmutableProptypes.default.listOf(_siteShape.default)
});

var _default = libraryShape;
exports.default = _default;
