// @bibliocommons/bc-prop-types/lib/components/listShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var listShape = _propTypes.default.shape({
  id: _propTypes.default.number,
  name: _propTypes.default.string,
  description: _propTypes.default.string,
  listType: _propTypes.default.oneOf(['topic_guide', 'other', 'genre_guide', 'read_a_like', 'k12_study_guide', 'health_guide', 'top_10', 'admin_newly_acquired', 'default_newly_acquired', 'admin_featured', 'staff_picks', 'teacher_picks', 'languages', 'personal_recommendation']),
  owner: _reactImmutableProptypes.default.mapContains({
    id: _propTypes.default.string
  }),
  imageUrl: _propTypes.default.string,
  imagePath: _propTypes.default.string,
  largeImagePath: _propTypes.default.string,
  created: _propTypes.default.string,
  updated: _propTypes.default.string,
  language: _propTypes.default.string,
  visibility: _propTypes.default.string,
  itemCount: _propTypes.default.number,
  isRanked: _propTypes.default.bool,
  isDraft: _propTypes.default.bool,
  agencyScope: _propTypes.default.string,
  intendedLocale: _propTypes.default.string,
  anchorBib: _reactImmutableProptypes.default.mapContains({
    id: _propTypes.default.string
  })
});

var _default = listShape;
exports.default = _default;
