// @bibliocommons/bc-prop-types/lib/components/listItemShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var jacketCoverShape = _reactImmutableProptypes.default.mapContains({
  type: _propTypes.default.string,
  smallCover: _propTypes.default.string,
  mediumCover: _propTypes.default.string,
  largeCover: _propTypes.default.string,
  localUrl: _propTypes.default.string
});

var bibListItemShape = _reactImmutableProptypes.default.mapContains({
  id: _propTypes.default.string,
  title: _propTypes.default.string,
  subtitle: _propTypes.default.string,
  authors: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  format: _propTypes.default.string,
  pubYear: _propTypes.default.string
});

var urlListItemShape = _reactImmutableProptypes.default.mapContains({
  url: _reactImmutableProptypes.default.mapContains({
    title: _propTypes.default.string,
    url: _propTypes.default.string
  })
});

var listItemShape = _reactImmutableProptypes.default.mapContains({
  id: _propTypes.default.number,
  type: _propTypes.default.string.isRequired,
  bibListItem: bibListItemShape,
  jacketCover: jacketCoverShape,
  urlListItem: urlListItemShape,
  annotation: _propTypes.default.string,
  isDraft: _propTypes.default.bool
});

var _default = listItemShape;
exports.default = _default;
