// @bibliocommons/bc-prop-types/lib/components/messagingShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var messagingShape = _reactImmutableProptypes.default.contains({
  message: _reactImmutableProptypes.default.mapContains({
    type: _propTypes.default.string.isRequired,
    id: _propTypes.default.string.isRequired,
    defaultMessage: _propTypes.default.string,
    values: _reactImmutableProptypes.default.map
  }),
  values: _reactImmutableProptypes.default.map,
  duration: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.bool]),
  fieldErrors: _reactImmutableProptypes.default.contains({
    classification: _propTypes.default.string,
    message: _propTypes.default.string,
    id: _propTypes.default.string,
    values: _reactImmutableProptypes.default.map
  }),
  failureDuration: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.bool]),
  useFormattedMessage: _propTypes.default.bool,
  actionButtonType: _propTypes.default.string,
  actionButtonProps: _reactImmutableProptypes.default.map
});

var _default = messagingShape;
exports.default = _default;
