// @bibliocommons/bc-prop-types/lib/components/stackMapShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var callNoShape = _reactImmutableProptypes.default.mapContains({
  start: _propTypes.default.string,
  end: _propTypes.default.string
});

var rangeShape = _reactImmutableProptypes.default.mapContains({
  callnoDisplay: _propTypes.default.string,
  callnos: _reactImmutableProptypes.default.listOf(callNoShape),
  coordinates: _propTypes.default.list,
  endcallno: _propTypes.default.string,
  label: _propTypes.default.string,
  rangename: _propTypes.default.string,
  rangeno: _propTypes.default.number,
  startcallno: _propTypes.default.string,
  x: _propTypes.default.number,
  y: _propTypes.default.number
});

var mapsShape = _reactImmutableProptypes.default.mapContains({
  directions: _propTypes.default.string,
  floorname: _propTypes.default.string,
  height: _propTypes.default.number,
  library: _propTypes.default.string,
  mapurl: _propTypes.default.string,
  ranges: _reactImmutableProptypes.default.listOf(rangeShape),
  width: _propTypes.default.number
});

var stackMapShape = _reactImmutableProptypes.default.mapContains({
  maps: _reactImmutableProptypes.default.listOf(mapsShape),
  callno: _propTypes.default.string,
  location: _propTypes.default.string,
  library: _propTypes.default.string
});

var _default = stackMapShape;
exports.default = _default;
