// @bibliocommons/bc-prop-types/lib/components/subscriptionShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var notesShape = _reactImmutableProptypes.default.mapContains({
  runNotes: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  holdingNotes: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  supplementaryNotes: _reactImmutableProptypes.default.listOf(_propTypes.default.string)
});

var subscriptionShape = _reactImmutableProptypes.default.mapContains({
  id: _propTypes.default.number.isRequired,
  collection: _propTypes.default.string,
  callNumbers: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  status: _propTypes.default.string,
  mediaType: _propTypes.default.string,
  notes: notesShape,
  location: _propTypes.default.string,
  runDuration: _reactImmutableProptypes.default.listOf(_reactImmutableProptypes.default.map)
});

var _default = subscriptionShape;
exports.default = _default;
