// @bibliocommons/bc-prop-types/lib/components/userIdentityShape.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactImmutableProptypes = _interopRequireDefault(require("react-immutable-proptypes"));

var userIdentityShape = _reactImmutableProptypes.default.contains({
  accounts: _reactImmutableProptypes.default.listOf(_propTypes.default.number),
  active: _propTypes.default.bool,
  avatar: _propTypes.default.string,
  encodedId: _propTypes.default.string,
  homeLibrary: _propTypes.default.string,
  homeLibraryFullName: _propTypes.default.string,
  homeLibrarySubdomain: _propTypes.default.string,
  homePatron: _propTypes.default.bool,
  id: _propTypes.default.number.isRequired,
  loggedIn: _propTypes.default.bool,
  name: _propTypes.default.string,
  permissions: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  preferences: _reactImmutableProptypes.default.map,
  roles: _reactImmutableProptypes.default.listOf(_propTypes.default.string),
  staff: _propTypes.default.bool,
  staffTakeover: _propTypes.default.bool,
  webUrl: _propTypes.default.string
});

var _default = userIdentityShape;
exports.default = _default;
