// @bibliocommons/deprecated-base-alerts/lib/components/dismiss/components/AlertDismiss.component.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = AlertDismiss;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _reactIntl = require("react-intl");

var _baseScreenReaderMessage = _interopRequireDefault(require("@bibliocommons/base-screen-reader-message"));

var _deprecatedBaseIcons = require("@bibliocommons/deprecated-base-icons");

require("../styles/AlertDismiss.scss");

function AlertDismiss(_ref) {
  var handleDismiss = _ref.handleDismiss;
  return /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: "close cp-deprecated-alert-dismiss",
    "data-key": "alert-dismiss",
    onClick: handleDismiss
  }, /*#__PURE__*/_react.default.createElement(_baseScreenReaderMessage.default, null, /*#__PURE__*/_react.default.createElement(_reactIntl.FormattedMessage, {
    id: "button_close"
  })), /*#__PURE__*/_react.default.createElement(_deprecatedBaseIcons.Cross, null));
}

AlertDismiss.propTypes = {
  handleDismiss: _propTypes.default.func.isRequired
};
